<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Epigenomic Analysis Package Built for R (epiRomics) • epiRomics</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Epigenomic Analysis Package Built for R (epiRomics)">
<meta property="og:description" content="A package designed to integrate various levels of epigenomic
    information, including, but not limited to ChIP and Histone
    next-generation sequencing. Regulatory network analysis can be done by
    using combinatory approaches to infer regions of significance, such as
    enhancers. Downstream analysis can identify co-occurance of epigenomic
    data located at regions of interest. Finally, this package allows for
    various results to be visualized.  This package is currently in
    development. Please contact &lt;ammawla@ucdavis.edu&gt; for suggestions,
    feedback, or bug reporting.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">epiRomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/getting-started-with-epiRomics.html">Getting Started with EpiRomics</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<div class="section level1">
<div class="page-header"><h1 id="epiromics-a-multi-omics-r-package-to-identify-and-visualize-enhancers">epiRomics: a multi-omics R package to identify and visualize enhancers<a class="anchor" aria-label="anchor" href="#epiromics-a-multi-omics-r-package-to-identify-and-visualize-enhancers"></a>
</h1></div>

<p>An R package designed to integrate and visualize various levels of epigenomic information, including but not limited to: ChIP, Histone, ATAC, and RNA sequencing. epiRomics is also designed for regulatory network analysis in order to identify enhancer and enhanceosome regions from these data.</p>
<p>This package is currently in development. Please contact <a href="mailto:ammawla@ucdavis.edu" class="email">ammawla@ucdavis.edu</a> for suggestions, feedback, or bug reporting.</p>
<p><strong>Paper:</strong> <a href="https://www.biorxiv.org/content/10.1101/2021.08.19.456732v1" class="external-link"><strong>epiRomics: a multi-omics R package to identify and visualize enhancers.</strong></a></p>
<p><strong>Resources</strong>:</p>
<p><a href="https://github.com/Huising-Lab/epiRomics/blob/main/vignettes/getting-started-with-epiromics.pdf" class="external-link">Getting Started with EpiRomics Walkthrough</a></p>
<p><a href="https://github.com/Huising-Lab/epiRomics/blob/main/doc/epiRomics_0.1.3.pdf" class="external-link">epiRomics Reference Manual</a></p>
<p><a href="https://www.biorxiv.org/content/biorxiv/early/2021/08/19/2021.08.19.456732/DC1/embed/media-1.docx?download=true" class="external-link">Methods for sample data in vignette</a></p>
<p><a href="https://www.biorxiv.org/content/biorxiv/early/2021/08/19/2021.08.19.456732/DC2/embed/media-2.xlsx?download=true" class="external-link">Accession table for sample data</a></p>
</div>
<div class="section level1">
<h1 id="package-installation-in-r">Package installation in R<a class="anchor" aria-label="anchor" href="#package-installation-in-r"></a>
</h1>
</div>
<div class="section level1">
<h1 id="install-devtools">Install Devtools<a class="anchor" aria-label="anchor" href="#install-devtools"></a>
</h1>
<p>install.packages(“devtools”)</p>
</div>
<div class="section level1">
<h1 id="install-epiromics">Install epiRomics<a class="anchor" aria-label="anchor" href="#install-epiromics"></a>
</h1>
<p>devtools::install_github(repo=“Huising-Lab/epiRomics”)</p>
</div>
<div class="section level1">
<h1 id="load-library">Load library<a class="anchor" aria-label="anchor" href="#load-library"></a>
</h1>
<p>library(epiRomics)</p>
</div>
<div class="section level1">
<h1 id="accessing-the-documentation">Accessing the documentation<a class="anchor" aria-label="anchor" href="#accessing-the-documentation"></a>
</h1>
<p><a href="https://github.com/Huising-Lab/epiRomics/blob/main/vignettes/getting-started-with-epiRomics.pdf" class="external-link">Getting Started with EpiRomics Walkthrough</a></p>
<p><a href="https://github.com/Huising-Lab/epiRomics/blob/main/doc/epiRomics_0.1.3.pdf" class="external-link">epiRomics Reference Manual</a></p>
<p>or</p>
<p>help(package = ‘epiRomics’, help_type = ‘html’)</p>
</div>
<div class="section level1">
<h1 id="vignette-data-notes">Vignette Data Notes<a class="anchor" aria-label="anchor" href="#vignette-data-notes"></a>
</h1>
<p>This package includes some example data to get you started, focusing on delineating putative human pancreatic islet enhancers between alpha and beta cells.</p>
<p>Human pancreatic islet alpha and beta ATAC- and companion RNA- Seq data were retrieved from GEO accession <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE76268" class="external-link">GSE76268</a> (Ackermann, et al., 2016). ATAC samples were processed using the <a href="https://github.com/ENCODE-DCC/atac-seq-pipeline" class="external-link">ENCODE-DCC ATAC sequencing pipeline</a>, aligning to the hg38 (Harrow, et al., 2012) build of the human genome (Consortium, 2012; Davis, et al., 2018). Peak calls generated through the pipeline using <a href="https://github.com/macs3-project/MACS" class="external-link">MACS2</a> (Zhang, et al., 2008) were analyzed downstream through the BioConductor package <a href="https://bioconductor.org/packages/release/bioc/html/DiffBind.html" class="external-link">DiffBind</a> (Ross-Innes, et al., 2012) in order to identify differentially enriched chromatin regions between the two cell types.</p>
<p>RNA samples were quality controlled using the tool <a href="https://github.com/OpenGene/fastp" class="external-link">Fastp</a> (Chen, et al., 2018), and aligned using <a href="https://github.com/alexdobin/STAR" class="external-link">s</a> (Dobin, et al., 2013) to the hg38 build of the human genome. Wiggle files produced by the <a href="https://github.com/alexdobin/STAR" class="external-link">STAR</a> aligner were then merged by cell type using <a href="https://github.com/ENCODE-DCC/kentUtils" class="external-link">UCSC command line tools</a> (Kent, et al., 2002)</p>
<p>All bigwigs were merged by cell type were subsetted to chromosome 1 using <a href="https://github.com/ENCODE-DCC/kentUtils" class="external-link">UCSC command line tools</a>.</p>
<p>ChIP-sequencing peak calls generated using <a href="https://github.com/macs3-project/MACS" class="external-link">MACS2</a> for human pancreatic islet transcription factors Foxa2, MafB, Nkx2.2, Nkx6.1, and Pdx1 were retrieved from the EMBL-EBI repository database <a href="https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-1919/" class="external-link">E-MTAB-1919</a> (Pasquali, et al., 2014). All peak calls were lifted over to the hg38 genome build using the <a href="http://genome.ucsc.edu/cgi-bin/hgLiftOver" class="external-link">UCSC genome browser liftOver tool</a> (Kent, et al., 2002).</p>
<p>Histone-sequencing peak calls generated using <a href="https://github.com/macs3-project/MACS" class="external-link">MACS2</a> for histones H3k27ac and H3k4me1 were retrieved from GEO accession <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE16256" class="external-link">GSE16256</a> (Bernstein, et al., 2010), and for histone H2A.Z from the EMBL-EBI repository database E-MTAB-1919](<a href="https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-1919/" class="external-link uri">https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-1919/</a>) (Pasquali, et al., 2014). All peak calls were lifted over to the hg38 genome build using the <a href="http://genome.ucsc.edu/cgi-bin/hgLiftOver" class="external-link">UCSC genome browser liftOver tool</a>.</p>
<p>The <a href="https://fantom.gsc.riken.jp/5/" class="external-link">FANTOM5 human enhancer database</a> (Lizio, et al., 2015) was retrieved, and all regions were lifted over to the hg38 genome build using the UCSC genome browser liftOver tool.</p>
<p>Human ultra-conserved non-coding elements (UCNEs) were retrieved form the <a href="https://ccg.epfl.ch/UCNEbase/" class="external-link">UCNE database</a> (Dimitrieva and Bucher, 2012), and all regions were lifted over to the hg38 genome build using the <a href="http://genome.ucsc.edu/cgi-bin/hgLiftOver" class="external-link">UCSC genome browser liftOver tool</a>.</p>
<p>The <a href="http://pasqualilab.upf.edu/app/isletregulome" class="external-link">human islet regulome database</a> was retrieved (Miguel-Escalada, et al., 2019) and all regions were lifted over to the hg38 genome build using the <a href="http://genome.ucsc.edu/cgi-bin/hgLiftOver" class="external-link">UCSC genome browser liftOver tool</a>.</p>
</div>
<div class="section level1">
<h1 id="citation-notes">Citation Notes<a class="anchor" aria-label="anchor" href="#citation-notes"></a>
</h1>
<p>If you use epiRomics, please cite:</p>
<p>Alex M. Mawla &amp; Mark O. Huising. (2021). <a href="https://www.biorxiv.org/content/10.1101/2021.08.19.456732v1" class="external-link">epiRomics: a multi-omics R package to identify and visualize enhancers.</a> bioRxiv 2021.08.19.456732</p>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>

<div class="community">
<h2 data-toc-skip>Community</h2>
<ul class="list-unstyled">
<li><a href="CODE_OF_CONDUCT.html">Code of conduct</a></li>
</ul>
</div>

<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing epiRomics</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Alex M. Mawla <br><small class="roles"> Author, maintainer </small>  </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://www.altmetric.com/details/112068712" class="external-link"><img src="https://img.shields.io/badge/Altmetric-13-green.svg"></a></li>
<li><a href="https://github.com/Huising-Lab/epiRomics/actions" class="external-link"><img src="https://github.com/Huising-Lab/epiRomics/workflows/R-CMD-check/badge.svg" alt="R build status"></a></li>
<li><a href="https://www.codefactor.io/repository/github/huising-lab/epiromics" class="external-link"><img src="https://www.codefactor.io/repository/github/huising-lab/epiromics/badge" alt="CodeFactor"></a></li>
<li><a href="https://codecov.io/gh/Huising-Lab/epiRomics?branch=main" class="external-link"><img src="https://codecov.io/gh/Huising-Lab/epiRomics/branch/main/graph/badge.svg" alt="Codecov test coverage"></a></li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Alex M. Mawla.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
