---
title: "Getting Started with EpiRomics"
#output: rmarkdown::html_vignette
author: "Authors: Alex M. Mawla & Mark O. Huising. Copyright 2020 - Present."
date: "`r Sys.Date()`"
output:
  prettydoc::html_pretty:
    toc: true
    theme: cayman
    highlight: github
  pdf_document:
    toc: true
    df_print: kable
vignette: >
 %\VignetteIndexEntry{Getting Started with EpiRomics}
 %\VignetteEngine{knitr::rmarkdown}
 %\VignetteEncoding{UTF-8}
 \usepackage[utf8]{inputenc}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  cache=TRUE,
  options(rmarkdown.html_vignette.check_title = FALSE, 
  tidy.opts = list(width.cutoff = 55), tidy = TRUE)
)
```

```{css, echo=FALSE}
pre code {
  white-space: pre-wrap;
}
```

# Abstract

**Summary** epiRomics is an R package designed to integrate multi-omics data in order to identify and visualize enhancer regions alongside gene expression and other epigenomic modifications. Regulatory network analysis can be done using combinatory approaches to infer regions of significance such as enhancers, when combining ChIP and histone data. Downstream analysis can identify co-occurrence of these regions of interest with other user-supplied data, such as chromatin availability or gene expression. Finally, this package allows for results to be visualized at high resolution in a stand-alone browser.

**Availability and Implementation** epiRomics is released under Artistic-2.0 License. The source code and documents are freely available through Github (<https://github.com/Huising-Lab/epiRomics>).

**Contact** ammawl\@ucdavis.edu or mhuising\@ucdavis.edu

**Supplementary information** Supplementary data, and methods are available online on *biorXiv* or *Github.*

**Competing Interest Statement**

The authors have declared no competing interest.

# Citation

If you use epiRomics in published research, please cite:

Mawla, AM& Huising, MO. **epiRomics: a multi-omics R package to identify and visualize enhancers.**

biorXiv 2021. doi:<https://doi.org/10.1101/2021.08.19.456732>

# Loading the epiRomics package and dependencies for vignette

```{r setup, tidy=TRUE}

## loading packages

library(epiRomics)

library(TxDb.Hsapiens.UCSC.hg38.knownGene)

library(org.Hs.eg.db)
```

# Brief explanation of example data

This package includes some example data to get you started, delineating human pancreatic islet enhancers between alpha and beta cells.

Human pancreatic islet alpha and beta ATAC- and companion RNA- Seq data were retrieved from GEO accession GSE76268 (Ackermann, et al., 2016).

ATAC samples were processed using the ENCODE-DCC ATAC sequencing pipeline, aligning to the hg38 (Harrow, et al., 2012) build of the human genome (Consortium, 2012; Davis, et al., 2018).

Peak calls generated through the pipeline using MACS2 (Zhang, et al., 2008) were analyzed downstream through the BioConductor package DiffBind (Ross-Innes, et al., 2012) in order to identify differentially enriched chromatin regions between the two cell types.

RNA samples were quality controlled using the tool fastp (Chen, et al., 2018), and aligned using STAR (Dobin, et al., 2013) to the hg38 build of the human genome. Wiggle files produced by the STAR aligner were then merged by cell type using UCSC command line tools.

Bigwigs merged by cell type were subsetted to chromosome 1 using UCSC command line tools (Kent, et al., 2010).

ChIP-sequencing peak calls generated using MACS2 for human pancreatic islet transcription factors Foxa2, MafB, Nkx2.2, Nkx6.1, and Pdx1 were retrieved from the EMBL-EBI repository database E-MTAB-1919 (Pasquali, et al., 2014). All peak calls were lifted over to the hg38 genome build using the UCSC genome browser liftOver tool (Kent, et al., 2002).

Histone-sequencing peak calls generated using MACS2 for histones H3k27ac and H3k4me1 were retrieved from GEO accession GSE16256 (Bernstein, et al., 2010), and for histone H2A.Z from the EMBL-EBI repository database E-MTAB-1919 (Pasquali, et al., 2014). All peak calls were lifted over to the hg38 genome build using the UCSC genome browser liftOver tool.

The FANTOM5 human enhancer database (Lizio, et al., 2015) was retrieved, and all regions were lifted over to the hg38 genome build using the UCSC genome browser liftOver tool.

Human ultra-conserved non-coding elements (UCNEs) were retrieved form the UCNE database (Dimitrieva and Bucher, 2012), and all regions were lifted over to the hg38 genome build using the UCSC genome browser liftOver tool.

The human islet regulome database was retrieved (Miguel-Escalada, et al., 2019) and all regions were lifted over to the hg38 genome build using the UCSC genome browser liftOver tool.

Lets load and take a look at how to properly format the datasets epiRomics uses to build the initial database.

```{r exampledatacheck, tidy=TRUE,   tidy.opts = list(width.cutoff = 70), fontsize=10}

## Required columns are: name, path, genome, format, and type

## The genome must also be in proper format, e.g. mm10 or hg38

## Type of data can be histone, methyl, SNP, or ChIP. ChIP is required for some downstream functions to work appropriately.

example_epiRomics_Db_sheet <- read.csv(file = system.file("extdata",
    "example_epiRomics_Db_sheet_user_paths.csv", package = "epiRomics"))

head(example_epiRomics_Db_sheet)
```

# How to load and build the database

```{r dataformatandload, tidy=TRUE, tidy.opts = list(width.cutoff = 70), fontsize=10}

## epiRomics_build_dB constructs a database of class epiRomics with this data sheet

epiRomics_dB <-
  epiRomics_build_dB(
    epiRomics_db_file = system.file(
      "extdata",
      "example_epiRomics_Db_sheet_user_paths.csv",
      package = "epiRomics"
    ),
    txdb_organism = "TxDb.Hsapiens.UCSC.hg38.knownGene::TxDb.Hsapiens.UCSC.hg38.knownGene",
    epiRomics_genome = "hg38",
    epiRomics_organism = "org.Hs.eg.db"
  )
```

# Delineating active enhancers using H3k4me1 and H3k27ac marks as a proxy

```{r putativeenhancers, tidy=TRUE,  tidy.opts = list(width.cutoff = 70), fontsize=10}
## Identifying active, putative enhancers

#3 There is a lot of flexibility for data exploration here. In this example, we search for putative enhancers using two histone marks known to co-occur at enhancer regions - h3k4me1 & h3k27ac

epiRomics_putative_enhancers <-
  epiRomics_enhancers(
    epiRomics_dB,
    epiRomics_histone_mark_1 = "h3k4me1",
    epiRomics_histone_mark_2 = "h3k27ac"
  )

## Taking a look, we see a  list of 19,692 putative enhancers demarked by H3k4me1 & H3k27ac

epiRomics_putative_enhancers@annotations
```

# Cross-referencing enhancer calls to other databases

## FANTOM Enhancer Database

```{r putativeenhancersfiltering, tidy=TRUE, tidy.opts = list(width.cutoff = 70), fontsize=10}
## Now we have a list of regions as possible candidates for enhancers, but where do we go from here? One way to increase confidence of these calls is to cross this list against an enhancer database, for instance, FANTOM. 

## NOTE: This option may not be available for all organisms.

epiRomics_putative_enhancers_filtered_fantom <-
  epiRomics_enhancers_filter(epiRomics_putative_enhancers, epiRomics_dB,
    epiRomics_type =
      "hg38_custom_fantom"
)

## Taking a look, we see a reduced number of 2,749 candidate regions

epiRomics_putative_enhancers_filtered_fantom@annotations
```

## Human Pancreatic Islet Regulome Enhancer Database

```{r putativeenhancersfilteringactiveregulome, tidy=TRUE, tidy.opts = list(width.cutoff = 70), fontsize=10}
## We can also filter putative enhancer calls against active enhancers from the human islet regulome database

epiRomics_putative_enhancers_filtered_regulome_active <-
  epiRomics_enhancers_filter(epiRomics_putative_enhancers, epiRomics_dB,
    epiRomics_type =
      "hg38_custom_regulome_active"
  )


epiRomics_putative_enhancers_filtered_regulome_active@annotations
```

## Human Pancreatic Islet Regulome Super-Enhancer Database

```{r putativeenhancersfilteringsuperregulome, tidy=TRUE, tidy.opts = list(width.cutoff = 70), fontsize=10}
## We can also filter putative enhancer calls against super enhancers from human islet regulome database

epiRomics_putative_enhancers_filtered_regulome_super <-
  epiRomics_enhancers_filter(epiRomics_putative_enhancers, epiRomics_dB,
    epiRomics_type =
      "hg38_custom_regulome_super"
  )

epiRomics_putative_enhancers_filtered_regulome_super@annotations
```

## Human Ultra-Conserved Non-Coding Elements Database

```{r putativeenhancersfilteringucnes, tidy=TRUE, tidy.opts = list(width.cutoff = 70), fontsize=10}
## We can also filter putative enhancer calls against Ultra-Conserved Non-Coding Elements

epiRomics_putative_enhancers_filtered_ucnes <-
  epiRomics_enhancers_filter(epiRomics_putative_enhancers, epiRomics_dB,
    epiRomics_type =
      "hg38_custom_ucnes"
  )


epiRomics_putative_enhancers_filtered_ucnes@annotations
```

# Screening for high transcription factor co-binding sites

Biology has established that enhancers can be quite redundant, and not all play an active role in regulating a cell's activity. How can we utilize other epigenomic data in order to identify true enhanceosome regions? One way is to cross this list against all ChIP data of the cell type. A true enhanceosome region should have made it through our filtering thus far, and contain several binding sites for known TFs. Co-binding is expected, and the list is sorted by the highest number of ChIP hits within the region.

```{r putativeenhancerstoenhanceosome, tidy=TRUE, tidy.opts = list(width.cutoff = 70), fontsize=10}

epiRomics_putative_enhanceosome_fantom <-
  epiRomics_enhanceosome(epiRomics_putative_enhancers_filtered_fantom, epiRomics_dB)

## Taking a look, we see the top candidates meet the criteria we list as expected

epiRomics_putative_enhanceosome_fantom@annotations


## Evaluate calls on chromosome 1
head(as.data.frame(epiRomics_putative_enhanceosome_fantom@annotations)[as.data.frame(epiRomics_putative_enhanceosome_fantom@annotations)$seqnames ==
  "chr1", ])

## Find Index
which(names(epiRomics_putative_enhanceosome_fantom@annotations) == 183)
```

# Transcription factor decision trees

ChIP dataset repositories are quite sizeable for many organisms and cell types, with the expectation to only grow larger. Many different TFs binding to a putative enhancer region may not be that meaningful in the context of your biological question. A further step would be to ask whether there are co-TFs that pop up together, and whether this pattern varies across the functional annotation of the genome, i.e. does the combination of two TFs on enhanceosomes change on the gene body compared to distal intergenic regions?

```{r fig1, fig.width=10, fig.height=10,  out.width="90%", fig.align="center", tidy.opts = list(width.cutoff = 70), fontsize=10}
 
plot(epiRomics_predictors(epiRomics_putative_enhanceosome_fantom))
```

# Intersecting and visualizing ATAC- and RNA-Seq data

What if you wanted to visualize co-binding on your FANTOM filtered putative enhancer region? And do you have additional data you want to include for visualization, such as ATAC and RNA Seq? Lets take a look at one of the top hits

```{r data, tidy=TRUE, tidy.opts = list(width.cutoff = 70), fontsize=10}

## Read in ATAC Seq and RNA Seq track bigwigs

## NOTE: These bigwigs are subsetted to chromosome 1. Indices not falling on chromosome 1 will return an error.

epiRomics_track_connection <- read.csv(
  system.file(
    "extdata",
    "example_epiRomics_BW_sheet_user_paths.csv",
    package = "epiRomics"
  )
)
```

```{r fig2, fig.width=10, fig.height=20, out.width="90%", fig.align="center", cache=TRUE, tidy.opts = list(width.cutoff = 70), fontsize=10}
epiRomics_track_layer_human(
  epiRomics_putative_enhanceosome_fantom,
  epiRomics_index = which(
    names(epiRomics_putative_enhanceosome_fantom@annotations) == 183
  ),
  epiRomics_dB = epiRomics_dB,
  epiRomics_track_connection = epiRomics_track_connection
)

```

```{r dataregulomeactivevisual, tidy=TRUE, tidy.opts = list(width.cutoff = 70), fontsize=10}
## What about a region that overlapped with active enhancers from the human islet regulome database?

epiRomics_putative_enhanceosome_regulome_active <-
  epiRomics_enhanceosome(
    epiRomics_putative_enhancers_filtered_regulome_active,
    epiRomics_dB
  )


epiRomics_putative_enhanceosome_regulome_active@annotations

## Evaluate calls on chromosome 1
head(as.data.frame(
  epiRomics_putative_enhanceosome_regulome_active@annotations
)[as.data.frame(epiRomics_putative_enhanceosome_regulome_active@annotations)$seqnames ==
  "chr1", ])

## Find Index
which(names(
  epiRomics_putative_enhanceosome_regulome_active@annotations
) == 82)
```

```{r fig3, fig.width=10, fig.height=20, out.width="90%", fig.align="center",cache=TRUE, tidy.opts = list(width.cutoff = 70), fontsize=10}

epiRomics_track_layer_human(
  epiRomics_putative_enhanceosome_regulome_active,
  epiRomics_index = which(
    names(
      epiRomics_putative_enhanceosome_regulome_active@annotations
    ) == 82
  ),
  epiRomics_dB = epiRomics_dB,
  epiRomics_track_connection = epiRomics_track_connection
)
```

```{r dataregulomesupervisual, tidy=TRUE, tidy.opts = list(width.cutoff = 70), fontsize=10}
## What about a region that overlapped with super enhancers from the human islet regulome database?

epiRomics_putative_enhanceosome_regulome_super <-
  epiRomics_enhanceosome(
    epiRomics_putative_enhancers_filtered_regulome_super,
    epiRomics_dB
  )


epiRomics_putative_enhanceosome_regulome_super@annotations


## Evaluate calls on chromosome 1
head(as.data.frame(epiRomics_putative_enhanceosome_regulome_super@annotations)[as.data.frame(epiRomics_putative_enhanceosome_regulome_super@annotations)$seqnames ==
  "chr1", ])

## Find Index
which(names(epiRomics_putative_enhanceosome_regulome_super@annotations) == 1)
```

```{r fig4, fig.width=10, fig.height=20,  out.width="90%", fig.align="center", cache=TRUE, tidy.opts = list(width.cutoff = 70), fontsize=10}
epiRomics_track_layer_human(
  epiRomics_putative_enhanceosome_regulome_super,
  epiRomics_index = which(
    names(epiRomics_putative_enhanceosome_regulome_super@annotations) == 1
  ),
  epiRomics_dB = epiRomics_dB,
  epiRomics_track_connection = epiRomics_track_connection
)
```

```{r dataucnes, tidy=TRUE, tidy.opts = list(width.cutoff = 70), fontsize=10}
## Or, about a region that overlapped with ultra-conserved non coding elements?

epiRomics_putative_enhanceosome_ucnes <-
  epiRomics_enhanceosome(epiRomics_putative_enhancers_filtered_ucnes, epiRomics_dB)


epiRomics_putative_enhanceosome_ucnes@annotations
```

```{r fig5, fig.width=10, fig.height=20,  out.width="90%", fig.align="center",cache=TRUE, tidy.opts = list(width.cutoff = 70), fontsize=10}
epiRomics_track_layer_human(
  epiRomics_putative_enhanceosome_ucnes,
  epiRomics_index = 9,
  epiRomics_dB = epiRomics_dB,
  epiRomics_track_connection = epiRomics_track_connection
)
```

```{r stringentfiltering, tidy.opts = list(width.cutoff = 70), fontsize=10}
## How about applying multiple filters to further increase the confidence of calls?

epiRomics_putative_enhancers_filtered_stringent <-
  epiRomics_enhancers_filter(
    epiRomics_enhancers_filter(
      epiRomics_enhancers_filter(
        epiRomics_enhancers_filter(epiRomics_putative_enhancers, epiRomics_dB,
          epiRomics_type =
            "hg38_custom_fantom"
        ),
        epiRomics_dB,
        epiRomics_type = "hg38_custom_regulome_active"
      ),
      epiRomics_dB,
      epiRomics_type = "hg38_custom_regulome_super"
    ),
    epiRomics_dB,
    epiRomics_type = "hg38_custom_ucnes"
  )


## Here, we see a highly conservative list of putative enhancer calls that overlap with four different functional annotations, suggesting the lowest hanging fruit for downstream bench-lab validation. NOTE: The UCNE database filter caused the greatest reduction in enhancer calls.
epiRomics_putative_enhancers_filtered_stringent@annotations


epiRomics_putative_enhanceosome_stringent <-
  epiRomics_enhanceosome(
    epiRomics_putative_enhancers_filtered_stringent,
    epiRomics_dB
  )
```

```{r fig6, fig.width=10, fig.height=20, out.width="90%", fig.align="center",cache=TRUE, tidy.opts = list(width.cutoff = 70), fontsize=10}
epiRomics_track_layer_human(
  epiRomics_putative_enhanceosome_stringent,
  epiRomics_index = 1,
  epiRomics_dB = epiRomics_dB,
  epiRomics_track_connection = epiRomics_track_connection
)
```

```{r integratechromatindata,  tidy=TRUE, tidy.opts = list(width.cutoff = 70), fontsize=10}
## How can we use these putative enhanceosome regions to infer biology between cell states? In this example, we will integrate ATAC-Seq data differential testing showing differences in chromatin accessibility between alpha and beta cells

## Read differentially binding data generated with DiffBind comparing human alpha and beta cell chromatin
b.v.a <-
  read.csv(system.file("extdata", "DBA_Beta_Versus_Alpha.csv", package = "epiRomics"))
b.v.a <- GRanges(b.v.a)

# Filter for beta enriched chromatin regions
beta.enriched <- b.v.a[b.v.a$Fold >= 1, ]

# Connect to our putative enhanceosomes
beta_enhancer_regions <-
  epiRomics_regions_of_interest(epiRomics_putative_enhanceosome_fantom, beta.enriched)
```

```{r fig7, fig.width=10, fig.height=20,  out.width="90%", fig.align="center",cache=TRUE}
## Now, lets visualize the top candidate region we found after connecting our differential chromatin analysis with the putative enhanceosomes
epiRomics_track_layer_human(
  beta_enhancer_regions,
  epiRomics_index = 1,
  epiRomics_dB = epiRomics_dB,
  epiRomics_track_connection = epiRomics_track_connection
)
```

# Session Information

Here is the output of `sessionInfo()` on the system on which this document was compiled:

```{r sessioninfo, tidy.opts = list(width.cutoff = 70), fontsize=10}
sessionInfo()
```
