---
title: "Getting Started with EpiRomics"
output: rmarkdown::html_vignette
author: "Authors: Alex M. Mawla. Copyright 2020 - Present."
vignette: >
%\VignetteIndexEntry{getting-started-with-epiRomics}
  %\VignetteEngine{knitr::knitr}
  %\VignetteEncoding{UTF-8}
  ---
  
  ```{r, include = FALSE}
knitr::opts_chunk$set(
collapse = TRUE,
comment = "#>"
)
```

```{r setup}
library(epiRomics)
```


```{r example data check}
# This package includes some example data to get you started.

system.file("extdata", "example_epiRomics_Db_sheet.csv", package = "epiRomics")

# Lets load and take a look at how to properly format the datasets epiRomics uses to build its database.

example_epiRomics_Db_sheet <-
  read.csv(file = system.file("extdata", "example_epiRomics_Db_sheet.csv", package = "epiRomics"))
```

```{r example data fix, include=FALSE}
example_epiRomics_Db_sheet$path <- paste0(system.file(package = "epiRomics"), "/extdata/", example_epiRomics_Db_sheet$path)
write.csv(example_epiRomics_Db_sheet, paste0(system.file(package = "epiRomics"), "/extdata/example_epiRomics_Db_sheet_user_paths.csv"), row.names = FALSE)
```

```{r data format and load}
# Required columns: name, path, genome, format, and type
# genome must be in proper format, e.g. mm10 or hg19
# type: histone or chip. chip is required for some downstream functions

head(example_epiRomics_Db_sheet)

# Database building
# epiRomics_build_dB constructs a database of class epiRomics with this data sheet

epiRomics_dB <-
  epiRomics_build_dB(
    epiRomics_db_file =  system.file(
      "extdata",
      "example_epiRomics_Db_sheet_user_paths.csv",
      package = "epiRomics"
    ),
    txdb_organism = "TxDb.Mmusculus.UCSC.mm10.knownGene::TxDb.Mmusculus.UCSC.mm10.knownGene",
    epiRomics_genome = "mm10",
    epiRomics_organism = "org.Mm.eg.db"
  )
```

```{r putative enhancers}
# Identifying putative enhancers
# There is a lot of flexibility for data exploration here. In this example, we search for putative enhancers using two histone marks known to co-occur at enhancer regions - h3k4me1 & h3k27ac

epiRomics_putative_enhancers <-
  epiRomics_enhancers(
    epiRomics_dB,
    epiRomics_histone_mark_1 = "h3k4me1",
    epiRomics_histone_mark_2 = "h3k27ac"
  )

# Taking a look, we see a sizeable list of 28,647 putative enhancers

epiRomics_putative_enhancers@annotations
```

```{r putative enhancers filtering}
# Now we have a list of regions as possible candidates for enhancers, but where do we go from here? One way to increase confidence of these calls is to cross this list against an enhancer database, for instance, FANTOM. NOTE: This option may not be available for all organisms 

epiRomics_putative_enhancers_filtered <-
  epiRomics_enhancers_filter(epiRomics_putative_enhancers, epiRomics_dB)

# Taking a look, we see a reduced number of 2,813 candidate regions  

epiRomics_putative_enhancers_filtered@annotations
```

```{r putative enhancers to enhanceosome}
# Biology has established that enhancers can be quite redundant, and not all play an active role in regulating a cell's activity. How can we utilize other epigenomic data in order to identify true enhanceosome regions? One way is to cross this list against all ChIP data of the cell type. A true enhanceosome region should have made it through our filtering thus far, and contain several binding sites for known TFs. Co-binding is expected, and the list is sorted by the highest number of ChIP hits within the region. 

epiRomics_putative_enhanceosome <-
  epiRomics_enhanceosome(epiRomics_putative_enhancers_filtered, epiRomics_dB)

# Taking a look, we see the top candidates meet the criteria we list as expected

epiRomics_putative_enhanceosome@annotations
```

```{r fig1, fig.width=10, fig.height=10}
# ChIP dataset repositories are quite sizeable for many organisms and cell types, with the expectation to only grow larger. Many different TFs binding to a putative enhancer region may not be that meaningful in the context of your biological question. A further step would be to ask whether there are co-TFs that pop up together, and whether this pattern varies across the functional annotation of the genome, i.e. does the combination of two TFs on enhanceosomes change on the gene body compared to distal intergenic regions? 

plot(epiRomics_predictors(epiRomics_putative_enhanceosome))
```

```{r fig2, fig.width=10, fig.height=15}
# What if you wanted to visualize co-binding on your putative enhancer region? And do you have additional data you want to include for visualization, such as ATAC Seq? Lets take a look at the top hit 

# Read in ATAC Seq. [can also be RNA Seq as well] 

epiRomics_track_connection <- read.csv(system.file("extdata", "example_epiRomics_BW_sheet.csv", package = "epiRomics"))

# Columns and information must be set as such

epiRomics_track_connection


epiRomics_track_layer(
  epiRomics_putative_enhanceosome,
  epiRomics_index = 1,
  epiRomics_dB = epiRomics_dB,
  epiRomics_track_connection = epiRomics_track_connection
)
```

```{r integrate chromatin data}
# How can we use these putative enhanceosome regions to infer biology between cell states? In this example, we will integrate ATAC-Seq data differential testing showing differences in chromatin accessibility between alpha and beta cells 

# Read differentially binding data generated with DiffBind comparing alpha and beta cell chromatin
a.v.b <- read.csv(system.file("extdata", "DBA_Alpha_Versus_Beta.csv", package = "epiRomics"))
a.v.b <- GRanges(a.v.b)

# Filter for beta enriched chromatin regions
beta.enriched <- a.v.b[a.v.b$Fold <= -1, ]

# Connect to our putative enhanceosomes
beta_enhancer_regions <- epiRomics_regions_of_interest(epiRomics_putative_enhanceosome, beta.enriched)
```

```{r fig3, fig.width=10, fig.height=15}
# Now, lets visualize the top candidate region we found after connecting our differential chromatin #analysis with the putative enhanceosomes 

epiRomics_track_layer(beta_enhancer_regions,
                      epiRomics_index = 1,
                      epiRomics_dB = epiRomics_dB,
                      epiRomics_track_connection = epiRomics_track_connection
)
```


