<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Getting Started with EpiRomics • epiRomics</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Getting Started with EpiRomics">
<meta property="og:description" content="epiRomics">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">epiRomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/getting-started-with-epiRomics.html">Getting Started with EpiRomics</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="getting-started-with-epiRomics_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Getting Started with EpiRomics</h1>
                        <h4 data-toc-skip class="author">Authors: Alex M. Mawla &amp; Mark O. Huising. Copyright 2020 - Present.</h4>
            
            <h4 data-toc-skip class="date">2021-12-06</h4>
      
      
      <div class="hidden name"><code>getting-started-with-epiRomics.Rmd</code></div>

    </div>

    
    
<style type="text/css">
pre code {
  white-space: pre-wrap;
}
</style>
<div class="section level1">
<h1 id="abstract">Abstract<a class="anchor" aria-label="anchor" href="#abstract"></a>
</h1>
<p><strong>Summary</strong> epiRomics is an R package designed to integrate multi-omics data in order to identify and visualize enhancer regions alongside gene expression and other epigenomic modifications. Regulatory network analysis can be done using combinatory approaches to infer regions of significance such as enhancers, when combining ChIP and histone data. Downstream analysis can identify co-occurrence of these regions of interest with other user-supplied data, such as chromatin availability or gene expression. Finally, this package allows for results to be visualized at high resolution in a stand-alone browser.</p>
<p><strong>Availability and Implementation</strong> epiRomics is released under Artistic-2.0 License. The source code and documents are freely available through Github (<a href="https://github.com/Huising-Lab/epiRomics" class="external-link uri">https://github.com/Huising-Lab/epiRomics</a>).</p>
<p><strong>Contact</strong> ammawl@ucdavis.edu or mhuising@ucdavis.edu</p>
<p><strong>Supplementary information</strong> Supplementary data, and methods are available online on <em>biorXiv</em> or <em>Github.</em></p>
<p><strong>Competing Interest Statement</strong></p>
<p>The authors have declared no competing interest.</p>
</div>
<div class="section level1">
<h1 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h1>
<p>If you use epiRomics in published research, please cite:</p>
<p>Mawla, AM&amp; Huising, MO. <strong>epiRomics: a multi-omics R package to identify and visualize enhancers.</strong></p>
<p>biorXiv 2021. doi:<a href="https://doi.org/10.1101/2021.08.19.456732" class="external-link uri">https://doi.org/10.1101/2021.08.19.456732</a></p>
</div>
<div class="section level1">
<h1 id="loading-the-epiromics-package-and-dependencies-for-vignette">Loading the epiRomics package and dependencies for vignette<a class="anchor" aria-label="anchor" href="#loading-the-epiromics-package-and-dependencies-for-vignette"></a>
</h1>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## loading packages</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">epiRomics</span><span class="op">)</span>
<span class="co">#&gt; epiRomics package loaded.</span>
<span class="co">#&gt; Automatic dependency check and verification of sample data presence</span>
<span class="co">#&gt; 'getOption("repos")' replaces Bioconductor standard repositories, see</span>
<span class="co">#&gt; '?repositories' for details</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; replacement repositories:</span>
<span class="co">#&gt;     CRAN: https://cloud.r-project.org</span>
<span class="co">#&gt; Bioconductor version 3.14 (BiocManager 1.30.16), R 4.1.2 (2021-11-01)</span>
<span class="co">#&gt; Old packages: 'enrichplot'</span>
<span class="co">#&gt; looking for  data.table</span>
<span class="co">#&gt; looking for  party</span>
<span class="co">#&gt; Loading required namespace: party</span>
<span class="co">#&gt; looking for  plyr</span>
<span class="co">#&gt; Loading required namespace: plyr</span>
<span class="co">#&gt; looking for  knitr</span>
<span class="co">#&gt; looking for  rmarkdown</span>
<span class="co">#&gt; looking for  igraph</span>
<span class="co">#&gt; Loading required namespace: igraph</span>
<span class="co">#&gt; looking for  AnnotationDbi</span>
<span class="co">#&gt; Loading required namespace: AnnotationDbi</span>
<span class="co">#&gt; looking for  annotatr</span>
<span class="co">#&gt; Loading required namespace: annotatr</span>
<span class="co">#&gt; looking for  BiocGenerics</span>
<span class="co">#&gt; looking for  GenomicFeatures</span>
<span class="co">#&gt; looking for  GenomicRanges</span>
<span class="co">#&gt; looking for  Gviz</span>
<span class="co">#&gt; Loading required namespace: Gviz</span>
<span class="co">#&gt; looking for  IRanges</span>
<span class="co">#&gt; looking for  rtracklayer</span>
<span class="co">#&gt; looking for  enrichplot</span>
<span class="co">#&gt; Loading required namespace: enrichplot</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; looking for  ChIPseeker</span>
<span class="co">#&gt; Loading required namespace: ChIPseeker</span>
<span class="co">#&gt; looking for  org.Hs.eg.db</span>
<span class="co">#&gt; Loading required namespace: org.Hs.eg.db</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; looking for  TxDb.Hsapiens.UCSC.hg38.knownGene</span>
<span class="co">#&gt; Loading required namespace: TxDb.Hsapiens.UCSC.hg38.knownGene</span>
<span class="co">#&gt; You are ready to go. For feedback, please email: ammawla@ucdavis.edu</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">TxDb.Hsapiens.UCSC.hg38.knownGene</span><span class="op">)</span>
<span class="co">#&gt; Loading required package: GenomicFeatures</span>
<span class="co">#&gt; Loading required package: BiocGenerics</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'BiocGenerics'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     IQR, mad, sd, var, xtabs</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     anyDuplicated, append, as.data.frame, basename, cbind, colnames,</span>
<span class="co">#&gt;     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,</span>
<span class="co">#&gt;     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,</span>
<span class="co">#&gt;     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span>
<span class="co">#&gt;     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,</span>
<span class="co">#&gt;     union, unique, unsplit, which.max, which.min</span>
<span class="co">#&gt; Loading required package: S4Vectors</span>
<span class="co">#&gt; Loading required package: stats4</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'S4Vectors'</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     expand.grid, I, unname</span>
<span class="co">#&gt; Loading required package: IRanges</span>
<span class="co">#&gt; Loading required package: GenomeInfoDb</span>
<span class="co">#&gt; Loading required package: GenomicRanges</span>
<span class="co">#&gt; Loading required package: AnnotationDbi</span>
<span class="co">#&gt; Loading required package: Biobase</span>
<span class="co">#&gt; Welcome to Bioconductor</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     Vignettes contain introductory material; view with</span>
<span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span>
<span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level1">
<h1 id="brief-explanation-of-example-data">Brief explanation of example data<a class="anchor" aria-label="anchor" href="#brief-explanation-of-example-data"></a>
</h1>
<p>This package includes some example data to get you started, delineating human pancreatic islet enhancers between alpha and beta cells.</p>
<p>Human pancreatic islet alpha and beta ATAC- and companion RNA- Seq data were retrieved from GEO accession GSE76268 (Ackermann, et al., 2016).</p>
<p>ATAC samples were processed using the ENCODE-DCC ATAC sequencing pipeline, aligning to the hg38 (Harrow, et al., 2012) build of the human genome (Consortium, 2012; Davis, et al., 2018).</p>
<p>Peak calls generated through the pipeline using MACS2 (Zhang, et al., 2008) were analyzed downstream through the BioConductor package DiffBind (Ross-Innes, et al., 2012) in order to identify differentially enriched chromatin regions between the two cell types.</p>
<p>RNA samples were quality controlled using the tool fastp (Chen, et al., 2018), and aligned using STAR (Dobin, et al., 2013) to the hg38 build of the human genome. Wiggle files produced by the STAR aligner were then merged by cell type using UCSC command line tools.</p>
<p>Bigwigs merged by cell type were subsetted to chromosome 1 using UCSC command line tools (Kent, et al., 2010).</p>
<p>ChIP-sequencing peak calls generated using MACS2 for human pancreatic islet transcription factors Foxa2, MafB, Nkx2.2, Nkx6.1, and Pdx1 were retrieved from the EMBL-EBI repository database E-MTAB-1919 (Pasquali, et al., 2014). All peak calls were lifted over to the hg38 genome build using the UCSC genome browser liftOver tool (Kent, et al., 2002).</p>
<p>Histone-sequencing peak calls generated using MACS2 for histones H3k27ac and H3k4me1 were retrieved from GEO accession GSE16256 (Bernstein, et al., 2010), and for histone H2A.Z from the EMBL-EBI repository database E-MTAB-1919 (Pasquali, et al., 2014). All peak calls were lifted over to the hg38 genome build using the UCSC genome browser liftOver tool.</p>
<p>The FANTOM5 human enhancer database (Lizio, et al., 2015) was retrieved, and all regions were lifted over to the hg38 genome build using the UCSC genome browser liftOver tool.</p>
<p>Human ultra-conserved non-coding elements (UCNEs) were retrieved form the UCNE database (Dimitrieva and Bucher, 2012), and all regions were lifted over to the hg38 genome build using the UCSC genome browser liftOver tool.</p>
<p>The human islet regulome database was retrieved (Miguel-Escalada, et al., 2019) and all regions were lifted over to the hg38 genome build using the UCSC genome browser liftOver tool.</p>
</div>
<div class="section level1">
<h1 id="how-to-load-and-build-the-database">How to load and build the database<a class="anchor" aria-label="anchor" href="#how-to-load-and-build-the-database"></a>
</h1>
<p>Lets load and take a look at how to properly format the datasets epiRomics uses to build the initial database.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">example_epiRomics_Db_sheet</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span>
  file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>
    <span class="st">"extdata"</span>,
    <span class="st">"example_epiRomics_Db_sheet_user_paths.csv"</span>,
    package <span class="op">=</span> <span class="st">"epiRomics"</span>
  <span class="op">)</span>
<span class="op">)</span>


<span class="co">## Required columns are: name, path, genome, format, and type</span>

<span class="co">## The genome must also be in proper format, e.g. mm10 or hg38</span>

<span class="co">## Type of data can be histone, methyl, SNP, or ChIP. </span>
<span class="co">## ChIP is required for some downstream functions to work appropriately.</span>

<span class="co">## Not run</span>
<span class="co">#head(example_epiRomics_Db_sheet)</span></code></pre></div>
<p>epiRomics_build_dB constructs a database of class epiRomics with this data sheet</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">epiRomics_dB</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/epiRomics_build_dB.html">epiRomics_build_dB</a></span><span class="op">(</span>
  epiRomics_db_file <span class="op">=</span>
    <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>
      <span class="st">"extdata"</span>,
      <span class="st">"example_epiRomics_Db_sheet_user_paths.csv"</span>,
      package <span class="op">=</span> <span class="st">"epiRomics"</span>
    <span class="op">)</span>,
  txdb_organism <span class="op">=</span>
    <span class="st">"TxDb.Hsapiens.UCSC.hg38.knownGene::TxDb.Hsapiens.UCSC.hg38.knownGene"</span>,
  epiRomics_genome <span class="op">=</span> <span class="st">"hg38"</span>,
  epiRomics_organism <span class="op">=</span> <span class="st">"org.Hs.eg.db"</span>
<span class="op">)</span>
<span class="co">#&gt; 'getOption("repos")' replaces Bioconductor standard repositories, see</span>
<span class="co">#&gt; '?repositories' for details</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; replacement repositories:</span>
<span class="co">#&gt;     CRAN: https://cloud.r-project.org</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; 'getOption("repos")' replaces Bioconductor standard repositories, see</span>
<span class="co">#&gt; '?repositories' for details</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; replacement repositories:</span>
<span class="co">#&gt;     CRAN: https://cloud.r-project.org</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; 'getOption("repos")' replaces Bioconductor standard repositories, see</span>
<span class="co">#&gt; '?repositories' for details</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; replacement repositories:</span>
<span class="co">#&gt;     CRAN: https://cloud.r-project.org</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; 'getOption("repos")' replaces Bioconductor standard repositories, see</span>
<span class="co">#&gt; '?repositories' for details</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; replacement repositories:</span>
<span class="co">#&gt;     CRAN: https://cloud.r-project.org</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; 'getOption("repos")' replaces Bioconductor standard repositories, see</span>
<span class="co">#&gt; '?repositories' for details</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; replacement repositories:</span>
<span class="co">#&gt;     CRAN: https://cloud.r-project.org</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; 'getOption("repos")' replaces Bioconductor standard repositories, see</span>
<span class="co">#&gt; '?repositories' for details</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; replacement repositories:</span>
<span class="co">#&gt;     CRAN: https://cloud.r-project.org</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; 'getOption("repos")' replaces Bioconductor standard repositories, see</span>
<span class="co">#&gt; '?repositories' for details</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; replacement repositories:</span>
<span class="co">#&gt;     CRAN: https://cloud.r-project.org</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; 'getOption("repos")' replaces Bioconductor standard repositories, see</span>
<span class="co">#&gt; '?repositories' for details</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; replacement repositories:</span>
<span class="co">#&gt;     CRAN: https://cloud.r-project.org</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; 'getOption("repos")' replaces Bioconductor standard repositories, see</span>
<span class="co">#&gt; '?repositories' for details</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; replacement repositories:</span>
<span class="co">#&gt;     CRAN: https://cloud.r-project.org</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; 'getOption("repos")' replaces Bioconductor standard repositories, see</span>
<span class="co">#&gt; '?repositories' for details</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; replacement repositories:</span>
<span class="co">#&gt;     CRAN: https://cloud.r-project.org</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; 'getOption("repos")' replaces Bioconductor standard repositories, see</span>
<span class="co">#&gt; '?repositories' for details</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; replacement repositories:</span>
<span class="co">#&gt;     CRAN: https://cloud.r-project.org</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; 'getOption("repos")' replaces Bioconductor standard repositories, see</span>
<span class="co">#&gt; '?repositories' for details</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; replacement repositories:</span>
<span class="co">#&gt;     CRAN: https://cloud.r-project.org</span>
<span class="co">#&gt; Building enhancers...</span>
<span class="co">#&gt; snapshotDate(): 2021-10-20</span>
<span class="co">#&gt; downloading 1 resources</span>
<span class="co">#&gt; retrieving 1 resource</span>
<span class="co">#&gt; loading from cache</span>
<span class="co">#&gt; require("rtracklayer")</span>
<span class="co">#&gt; 'getOption("repos")' replaces Bioconductor standard repositories, see</span>
<span class="co">#&gt; '?repositories' for details</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; replacement repositories:</span>
<span class="co">#&gt;     CRAN: https://cloud.r-project.org</span>
<span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span>
<span class="co">#&gt; Building promoters...</span>
<span class="co">#&gt; Building 1to5kb upstream of TSS...</span>
<span class="co">#&gt; Building intergenic...</span>
<span class="co">#&gt; Building cds...</span>
<span class="co">#&gt; Building 5UTRs...</span>
<span class="co">#&gt; Building 3UTRs...</span>
<span class="co">#&gt; Building exons...</span>
<span class="co">#&gt; Building first exons...</span>
<span class="co">#&gt; Building introns...</span>
<span class="co">#&gt; Building intron exon boundaries...</span>
<span class="co">#&gt; Building exon intron boundaries...</span>
<span class="co">#&gt; Building CpG islands...</span>
<span class="co">#&gt; Building CpG shores...</span>
<span class="co">#&gt; Building CpG shelves...</span>
<span class="co">#&gt; Building inter-CpG-islands...</span>
<span class="co">#&gt; snapshotDate(): 2021-10-20</span>
<span class="co">#&gt; Building lncRNA transcripts...</span>
<span class="co">#&gt; downloading 1 resources</span>
<span class="co">#&gt; retrieving 1 resource</span>
<span class="co">#&gt; loading from cache</span></code></pre></div>
</div>
<div class="section level1">
<h1 id="delineating-active-enhancers-using-h3k4me1-and-h3k27ac-marks-as-a-proxy">Delineating active enhancers using H3k4me1 and H3k27ac marks as a proxy<a class="anchor" aria-label="anchor" href="#delineating-active-enhancers-using-h3k4me1-and-h3k27ac-marks-as-a-proxy"></a>
</h1>
<p>There is a lot of flexibility for data exploration here. In this example, we search for putative enhancers using two histone marks known to co-occur at enhancer regions - h3k4me1 &amp; h3k27ac</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">epiRomics_putative_enhancers</span> <span class="op">&lt;-</span>
  <span class="fu"><a href="../reference/epiRomics_enhancers.html">epiRomics_enhancers</a></span><span class="op">(</span>
    <span class="va">epiRomics_dB</span>,
    epiRomics_histone_mark_1 <span class="op">=</span>
      <span class="st">"h3k4me1"</span>,
    epiRomics_histone_mark_2 <span class="op">=</span> <span class="st">"h3k27ac"</span>
  <span class="op">)</span>

<span class="co">## Taking a look, we see a  list of 19,692 putative enhancers </span>

<span class="va">epiRomics_putative_enhancers</span><span class="op">@</span><span class="va">annotations</span>
<span class="co">#&gt; GRanges object with 19692 ranges and 0 metadata columns:</span>
<span class="co">#&gt;           seqnames            ranges strand</span>
<span class="co">#&gt;              &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;       [1]     chr1    999886-1000011      *</span>
<span class="co">#&gt;       [2]     chr1   1000228-1000811      *</span>
<span class="co">#&gt;       [3]     chr1   1000850-1001468      *</span>
<span class="co">#&gt;       [4]     chr1   1005007-1006023      *</span>
<span class="co">#&gt;       [5]     chr1   1013701-1013893      *</span>
<span class="co">#&gt;       ...      ...               ...    ...</span>
<span class="co">#&gt;   [19688]     chrY 12392544-12392994      *</span>
<span class="co">#&gt;   [19689]     chrY 13282680-13282760      *</span>
<span class="co">#&gt;   [19690]     chrY 15455449-15455788      *</span>
<span class="co">#&gt;   [19691]     chrY 19066496-19066508      *</span>
<span class="co">#&gt;   [19692]     chrY 19075542-19075899      *</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 640 sequences (1 circular) from hg38 genome</span></code></pre></div>
</div>
<div class="section level1">
<h1 id="cross-referencing-enhancer-calls-to-other-databases">Cross-referencing enhancer calls to other databases<a class="anchor" aria-label="anchor" href="#cross-referencing-enhancer-calls-to-other-databases"></a>
</h1>
<div class="section level2">
<h2 id="fantom-enhancer-database">FANTOM Enhancer Database<a class="anchor" aria-label="anchor" href="#fantom-enhancer-database"></a>
</h2>
<p>Now we have a list of regions as possible candidates for enhancers, but where do we go from here? One way to increase confidence of these calls is to cross this list against an enhancer database, for instance, FANTOM.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## NOTE: This option may not be available for all organisms.</span>

<span class="va">epiRomics_putative_enhancers_filtered_fantom</span> <span class="op">&lt;-</span>
  <span class="fu"><a href="../reference/epiRomics_enhancers_filter.html">epiRomics_enhancers_filter</a></span><span class="op">(</span><span class="va">epiRomics_putative_enhancers</span>, <span class="va">epiRomics_dB</span>,
                             epiRomics_type <span class="op">=</span>
                               <span class="st">"hg38_custom_fantom"</span><span class="op">)</span>

<span class="co">## Taking a look, we see a reduced number of 2,749 candidate regions</span>

<span class="va">epiRomics_putative_enhancers_filtered_fantom</span><span class="op">@</span><span class="va">annotations</span>
<span class="co">#&gt; GRanges object with 2749 ranges and 0 metadata columns:</span>
<span class="co">#&gt;          seqnames              ranges strand</span>
<span class="co">#&gt;             &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;      [1]     chr1     1021242-1021277      *</span>
<span class="co">#&gt;      [2]     chr1     1021318-1021698      *</span>
<span class="co">#&gt;      [3]     chr1     1079632-1080061      *</span>
<span class="co">#&gt;      [4]     chr1     1080101-1080628      *</span>
<span class="co">#&gt;      [5]     chr1     1128200-1128445      *</span>
<span class="co">#&gt;      ...      ...                 ...    ...</span>
<span class="co">#&gt;   [2745]     chrX 154369950-154370183      *</span>
<span class="co">#&gt;   [2746]     chrX 154371971-154372237      *</span>
<span class="co">#&gt;   [2747]     chrX 154372350-154372695      *</span>
<span class="co">#&gt;   [2748]     chrX 154517139-154517596      *</span>
<span class="co">#&gt;   [2749]     chrX 154734550-154734738      *</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 640 sequences (1 circular) from hg38 genome</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="human-pancreatic-islet-regulome-enhancer-database">Human Pancreatic Islet Regulome Enhancer Database<a class="anchor" aria-label="anchor" href="#human-pancreatic-islet-regulome-enhancer-database"></a>
</h2>
<p>We can also filter putative enhancer calls against active enhancers from the human islet regulome database</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">epiRomics_putative_enhancers_filtered_regulome_active</span> <span class="op">&lt;-</span>
  <span class="fu"><a href="../reference/epiRomics_enhancers_filter.html">epiRomics_enhancers_filter</a></span><span class="op">(</span><span class="va">epiRomics_putative_enhancers</span>,
                             <span class="va">epiRomics_dB</span>,
                             epiRomics_type <span class="op">=</span> <span class="st">"hg38_custom_regulome_active"</span><span class="op">)</span>


<span class="va">epiRomics_putative_enhancers_filtered_regulome_active</span><span class="op">@</span><span class="va">annotations</span>
<span class="co">#&gt; GRanges object with 6025 ranges and 0 metadata columns:</span>
<span class="co">#&gt;          seqnames              ranges strand</span>
<span class="co">#&gt;             &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;      [1]     chr1     1068896-1068951      *</span>
<span class="co">#&gt;      [2]     chr1     1069171-1069333      *</span>
<span class="co">#&gt;      [3]     chr1     1079632-1080061      *</span>
<span class="co">#&gt;      [4]     chr1     1080101-1080628      *</span>
<span class="co">#&gt;      [5]     chr1     1158358-1158930      *</span>
<span class="co">#&gt;      ...      ...                 ...    ...</span>
<span class="co">#&gt;   [6021]     chrX 153381411-153381523      *</span>
<span class="co">#&gt;   [6022]     chrX 153381677-153381956      *</span>
<span class="co">#&gt;   [6023]     chrX 153382322-153382448      *</span>
<span class="co">#&gt;   [6024]     chrX 153985442-153985689      *</span>
<span class="co">#&gt;   [6025]     chrX 154091801-154091996      *</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 640 sequences (1 circular) from hg38 genome</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="human-pancreatic-islet-regulome-super-enhancer-database">Human Pancreatic Islet Regulome Super-Enhancer Database<a class="anchor" aria-label="anchor" href="#human-pancreatic-islet-regulome-super-enhancer-database"></a>
</h2>
<p>We can also filter putative enhancer calls against super enhancers from human islet regulome database</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">epiRomics_putative_enhancers_filtered_regulome_super</span> <span class="op">&lt;-</span>
  <span class="fu"><a href="../reference/epiRomics_enhancers_filter.html">epiRomics_enhancers_filter</a></span><span class="op">(</span><span class="va">epiRomics_putative_enhancers</span>,
                             <span class="va">epiRomics_dB</span>, epiRomics_type <span class="op">=</span> <span class="st">"hg38_custom_regulome_super"</span><span class="op">)</span>

<span class="va">epiRomics_putative_enhancers_filtered_regulome_super</span><span class="op">@</span><span class="va">annotations</span>
<span class="co">#&gt; GRanges object with 2401 ranges and 0 metadata columns:</span>
<span class="co">#&gt;          seqnames            ranges strand</span>
<span class="co">#&gt;             &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;      [1]     chr1   7574092-7574479      *</span>
<span class="co">#&gt;      [2]     chr1   7574640-7575094      *</span>
<span class="co">#&gt;      [3]     chr1   8169274-8169689      *</span>
<span class="co">#&gt;      [4]     chr1   8170112-8170857      *</span>
<span class="co">#&gt;      [5]     chr1   8174089-8174358      *</span>
<span class="co">#&gt;      ...      ...               ...    ...</span>
<span class="co">#&gt;   [2397]    chr22 46109916-46110442      *</span>
<span class="co">#&gt;   [2398]    chr22 46115774-46116154      *</span>
<span class="co">#&gt;   [2399]    chr22 46116326-46116501      *</span>
<span class="co">#&gt;   [2400]     chrX 39813348-39813627      *</span>
<span class="co">#&gt;   [2401]     chrX 39814304-39814607      *</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 640 sequences (1 circular) from hg38 genome</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="human-ultra-conserved-non-coding-elements-database">Human Ultra-Conserved Non-Coding Elements Database<a class="anchor" aria-label="anchor" href="#human-ultra-conserved-non-coding-elements-database"></a>
</h2>
<p>We can also filter putative enhancer calls against Ultra-Conserved Non-Coding Elements</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">epiRomics_putative_enhancers_filtered_ucnes</span> <span class="op">&lt;-</span>
  <span class="fu"><a href="../reference/epiRomics_enhancers_filter.html">epiRomics_enhancers_filter</a></span><span class="op">(</span><span class="va">epiRomics_putative_enhancers</span>,
                             <span class="va">epiRomics_dB</span>,
                             epiRomics_type <span class="op">=</span> <span class="st">"hg38_custom_ucnes"</span><span class="op">)</span>

<span class="va">epiRomics_putative_enhancers_filtered_ucnes</span><span class="op">@</span><span class="va">annotations</span>
<span class="co">#&gt; GRanges object with 11 ranges and 0 metadata columns:</span>
<span class="co">#&gt;        seqnames              ranges strand</span>
<span class="co">#&gt;           &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;    [1]     chr1 164635220-164635921      *</span>
<span class="co">#&gt;    [2]     chr1 164711914-164712296      *</span>
<span class="co">#&gt;    [3]     chr1 164712350-164713071      *</span>
<span class="co">#&gt;    [4]     chr1 200079185-200079426      *</span>
<span class="co">#&gt;    [5]     chr1 213585694-213586385      *</span>
<span class="co">#&gt;    [6]     chr3   71131859-71132164      *</span>
<span class="co">#&gt;    [7]     chr9 106921420-106921764      *</span>
<span class="co">#&gt;    [8]    chr11 114163425-114164860      *</span>
<span class="co">#&gt;    [9]    chr15   36903894-36904085      *</span>
<span class="co">#&gt;   [10]    chr15   53447393-53447809      *</span>
<span class="co">#&gt;   [11]    chr21   16534340-16534665      *</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 640 sequences (1 circular) from hg38 genome</span></code></pre></div>
</div>
</div>
<div class="section level1">
<h1 id="screening-for-high-transcription-factor-co-binding-sites">Screening for high transcription factor co-binding sites<a class="anchor" aria-label="anchor" href="#screening-for-high-transcription-factor-co-binding-sites"></a>
</h1>
<p>Biology has established that enhancers can be quite redundant, and not all play an active role in regulating a cell’s activity. How can we utilize other epigenomic data in order to identify true enhanceosome regions? One way is to cross this list against all ChIP data of the cell type. A true enhanceosome region should have made it through our filtering thus far, and contain several binding sites for known TFs. Co-binding is expected, and the list is sorted by the highest number of ChIP hits within the region.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">epiRomics_putative_enhanceosome_fantom</span> <span class="op">&lt;-</span>
  <span class="fu"><a href="../reference/epiRomics_enhanceosome.html">epiRomics_enhanceosome</a></span><span class="op">(</span><span class="va">epiRomics_putative_enhancers_filtered_fantom</span>,
                         <span class="va">epiRomics_dB</span><span class="op">)</span>
<span class="co">#&gt; &gt;&gt; preparing features information...      2021-12-06 02:35:52 </span>
<span class="co">#&gt; &gt;&gt; identifying nearest features...        2021-12-06 02:35:53 </span>
<span class="co">#&gt; &gt;&gt; calculating distance from peak to TSS...   2021-12-06 02:35:54 </span>
<span class="co">#&gt; &gt;&gt; assigning genomic annotation...        2021-12-06 02:35:54 </span>
<span class="co">#&gt; &gt;&gt; adding gene annotation...          2021-12-06 02:36:36</span>
<span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span>
<span class="co">#&gt; &gt;&gt; assigning chromosome lengths           2021-12-06 02:36:36 </span>
<span class="co">#&gt; &gt;&gt; done...                    2021-12-06 02:36:36</span>

<span class="co">## Taking a look, we see the top candidates meet the criteria we list as expected</span>

<span class="va">epiRomics_putative_enhanceosome_fantom</span><span class="op">@</span><span class="va">annotations</span>
<span class="co">#&gt; GRanges object with 2749 ranges and 19 metadata columns:</span>
<span class="co">#&gt;        seqnames              ranges strand |     foxa2      mafb    nkx2_2</span>
<span class="co">#&gt;           &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;</span>
<span class="co">#&gt;    183     chr1 154418514-154419684      * |         2         2         1</span>
<span class="co">#&gt;   1096     chr9     2242369-2242873      * |         2         1         1</span>
<span class="co">#&gt;   2615    chr22   30310745-30311570      * |         2         1         2</span>
<span class="co">#&gt;     34     chr1   10685395-10688670      * |         1         0         1</span>
<span class="co">#&gt;    792     chr6   30748438-30749427      * |         2         1         1</span>
<span class="co">#&gt;    ...      ...                 ...    ... .       ...       ...       ...</span>
<span class="co">#&gt;   2743     chrX 153927339-153927701      * |         0         0         0</span>
<span class="co">#&gt;   2745     chrX 154369950-154370183      * |         0         0         0</span>
<span class="co">#&gt;   2746     chrX 154371971-154372237      * |         0         0         0</span>
<span class="co">#&gt;   2747     chrX 154372350-154372695      * |         0         0         0</span>
<span class="co">#&gt;   2748     chrX 154517139-154517596      * |         0         0         0</span>
<span class="co">#&gt;           nxk6_1      pdx1      h2az ChIP_Hits             annotation   geneChr</span>
<span class="co">#&gt;        &lt;integer&gt; &lt;integer&gt; &lt;integer&gt; &lt;numeric&gt;            &lt;character&gt; &lt;integer&gt;</span>
<span class="co">#&gt;    183         2         1         2        10 Intron (ENST00000622..         1</span>
<span class="co">#&gt;   1096         1         2         1         8      Distal Intergenic         9</span>
<span class="co">#&gt;   2615         1         1         1         8       Promoter (2-3kb)        22</span>
<span class="co">#&gt;     34         1         2         2         7 Intron (ENST00000377..         1</span>
<span class="co">#&gt;    792         1         1         1         7 Intron (ENST00000656..         6</span>
<span class="co">#&gt;    ...       ...       ...       ...       ...                    ...       ...</span>
<span class="co">#&gt;   2743         0         0         0         0       Promoter (&lt;=1kb)        23</span>
<span class="co">#&gt;   2745         0         0         0         0       Promoter (1-2kb)        23</span>
<span class="co">#&gt;   2746         0         0         0         0       Promoter (&lt;=1kb)        23</span>
<span class="co">#&gt;   2747         0         0         0         0       Promoter (1-2kb)        23</span>
<span class="co">#&gt;   2748         0         0         0         0       Promoter (&lt;=1kb)        23</span>
<span class="co">#&gt;        geneStart   geneEnd geneLength geneStrand      geneId      transcriptId</span>
<span class="co">#&gt;        &lt;integer&gt; &lt;integer&gt;  &lt;integer&gt;  &lt;integer&gt; &lt;character&gt;       &lt;character&gt;</span>
<span class="co">#&gt;    183 154429343 154449979      20637          1        3570 ENST00000476006.5</span>
<span class="co">#&gt;   1096   2181571   2186183       4613          1        6595 ENST00000635392.1</span>
<span class="co">#&gt;   2615  30292008  30307890      15883          2       83874 ENST00000403362.5</span>
<span class="co">#&gt;     34  10660737  10693912      33176          2       54897 ENST00000478728.2</span>
<span class="co">#&gt;    792  30743199  30744547       1349          2        8870 ENST00000259874.6</span>
<span class="co">#&gt;    ...       ...       ...        ...        ...         ...               ...</span>
<span class="co">#&gt;   2743 153920715 153926860       6146          2         393 ENST00000422091.1</span>
<span class="co">#&gt;   2745 154348524 154371203      22680          2        2316 ENST00000420627.5</span>
<span class="co">#&gt;   2746 154348529 154371283      22755          2        2316 ENST00000422373.6</span>
<span class="co">#&gt;   2747 154348529 154371283      22755          2        2316 ENST00000422373.6</span>
<span class="co">#&gt;   2748 154506204 154516242      10039          2       60343 ENST00000434658.6</span>
<span class="co">#&gt;        distanceToTSS         ENSEMBL      SYMBOL               GENENAME</span>
<span class="co">#&gt;            &lt;numeric&gt;     &lt;character&gt; &lt;character&gt;            &lt;character&gt;</span>
<span class="co">#&gt;    183         -9659 ENSG00000160712        IL6R interleukin 6 receptor</span>
<span class="co">#&gt;   1096         60798 ENSG00000080503     SMARCA2 SWI/SNF related, mat..</span>
<span class="co">#&gt;   2615         -2855 ENSG00000099992    TBC1D10A TBC1 domain family m..</span>
<span class="co">#&gt;     34          5242 ENSG00000130940       CASZ1   castor zinc finger 1</span>
<span class="co">#&gt;    792         -3891 ENSG00000137331        IER3 immediate early resp..</span>
<span class="co">#&gt;    ...           ...             ...         ...                    ...</span>
<span class="co">#&gt;   2743          -479 ENSG00000089820     ARHGAP4 Rho GTPase activatin..</span>
<span class="co">#&gt;   2745          1020 ENSG00000196924        FLNA              filamin A</span>
<span class="co">#&gt;   2746          -688 ENSG00000196924        FLNA              filamin A</span>
<span class="co">#&gt;   2747         -1067 ENSG00000196924        FLNA              filamin A</span>
<span class="co">#&gt;   2748          -897 ENSG00000071889       FAM3A FAM3 metabolism regu..</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 640 sequences (1 circular) from hg38 genome</span>

<span class="co">## Evaluate calls on chromosome 1</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span>
  <span class="va">epiRomics_putative_enhanceosome_fantom</span><span class="op">@</span><span class="va">annotations</span>
  <span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">epiRomics_putative_enhanceosome_fantom</span><span class="op">@</span><span class="va">annotations</span><span class="op">)</span><span class="op">$</span><span class="va">seqnames</span> 
    <span class="op">==</span> <span class="st">"chr1"</span>,<span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;     seqnames     start       end width strand foxa2 mafb nkx2_2 nxk6_1 pdx1</span>
<span class="co">#&gt; 183     chr1 154418514 154419684  1171      *     2    2      1      2    1</span>
<span class="co">#&gt; 34      chr1  10685395  10688670  3276      *     1    0      1      1    2</span>
<span class="co">#&gt; 24      chr1   8170112   8170857   746      *     1    1      1      1    1</span>
<span class="co">#&gt; 67      chr1  21638834  21639978  1145      *     1    1      1      1    1</span>
<span class="co">#&gt; 71      chr1  22414515  22414806   292      *     1    1      1      1    1</span>
<span class="co">#&gt; 256     chr1 205318810 205319660   851      *     1    1      2      1    0</span>
<span class="co">#&gt;     h2az ChIP_Hits                                       annotation geneChr</span>
<span class="co">#&gt; 183    2        10   Intron (ENST00000622330.4/3570, intron 1 of 6)       1</span>
<span class="co">#&gt; 34     2         7 Intron (ENST00000377022.8/54897, intron 4 of 20)       1</span>
<span class="co">#&gt; 24     1         6                                Distal Intergenic       1</span>
<span class="co">#&gt; 67     1         6  Intron (ENST00000290101.8/5909, intron 2 of 26)       1</span>
<span class="co">#&gt; 71     1         6                                Distal Intergenic       1</span>
<span class="co">#&gt; 256    1         6                                 Promoter (2-3kb)       1</span>
<span class="co">#&gt;     geneStart   geneEnd geneLength geneStrand    geneId      transcriptId</span>
<span class="co">#&gt; 183 154429343 154449979      20637          1      3570 ENST00000476006.5</span>
<span class="co">#&gt; 34   10660737  10693912      33176          2     54897 ENST00000478728.2</span>
<span class="co">#&gt; 24    8201518   8215207      13690          1 102724539 ENST00000670361.1</span>
<span class="co">#&gt; 67   21596221  21651820      55600          2      5909 ENST00000471600.6</span>
<span class="co">#&gt; 71   22428838  22511763      82926          1      9923 ENST00000650433.1</span>
<span class="co">#&gt; 256 205302063 205321745      19683          2     81788 ENST00000367157.6</span>
<span class="co">#&gt;     distanceToTSS         ENSEMBL    SYMBOL</span>
<span class="co">#&gt; 183         -9659 ENSG00000160712      IL6R</span>
<span class="co">#&gt; 34           5242 ENSG00000130940     CASZ1</span>
<span class="co">#&gt; 24         -30661 ENSG00000227634 LINC01714</span>
<span class="co">#&gt; 67          11842 ENSG00000076864   RAP1GAP</span>
<span class="co">#&gt; 71         -14032 ENSG00000184677    ZBTB40</span>
<span class="co">#&gt; 256          2085 ENSG00000163545     NUAK2</span>
<span class="co">#&gt;                                        GENENAME</span>
<span class="co">#&gt; 183                      interleukin 6 receptor</span>
<span class="co">#&gt; 34                         castor zinc finger 1</span>
<span class="co">#&gt; 24  long intergenic non-protein coding RNA 1714</span>
<span class="co">#&gt; 67               RAP1 GTPase activating protein</span>
<span class="co">#&gt; 71     zinc finger and BTB domain containing 40</span>
<span class="co">#&gt; 256                        NUAK family kinase 2</span>

<span class="co">## Find Index</span>

<span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">epiRomics_putative_enhanceosome_fantom</span><span class="op">@</span><span class="va">annotations</span><span class="op">)</span> <span class="op">==</span> <span class="fl">183</span><span class="op">)</span>
<span class="co">#&gt; [1] 1</span></code></pre></div>
</div>
<div class="section level1">
<h1 id="transcription-factor-decision-trees">Transcription factor decision trees<a class="anchor" aria-label="anchor" href="#transcription-factor-decision-trees"></a>
</h1>
<p>ChIP dataset repositories are quite sizeable for many organisms and cell types, with the expectation to only grow larger. Many different TFs binding to a putative enhancer region may not be that meaningful in the context of your biological question. A further step would be to ask whether there are co-TFs that pop up together, and whether this pattern varies across the functional annotation of the genome, i.e. does the combination of two TFs on enhanceosomes change on the gene body compared to distal intergenic regions?</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"> 
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="../reference/epiRomics_predictors.html">epiRomics_predictors</a></span><span class="op">(</span><span class="va">epiRomics_putative_enhanceosome_fantom</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="getting-started-with-epiRomics_files/figure-html/fig1-1.png" width="100%" style="display: block; margin: auto;"></p>
</div>
<div class="section level1">
<h1 id="intersecting-and-visualizing-atac--and-rna-seq-data">Intersecting and visualizing ATAC- and RNA-Seq data<a class="anchor" aria-label="anchor" href="#intersecting-and-visualizing-atac--and-rna-seq-data"></a>
</h1>
<p>What if you wanted to visualize co-binding on your FANTOM filtered putative enhancer region? And do you have additional data you want to include for visualization, such as ATAC and RNA Seq? Lets take a look at one of the top hits</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Read in ATAC Seq and RNA Seq track bigwigs</span>

<span class="co">## NOTE: These bigwigs are subsetted to chromosome 1. </span>
<span class="co">## Indices not falling on chromosome 1 will return an error.</span>

<span class="va">epiRomics_track_connection</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>
    <span class="st">"extdata"</span>,
    <span class="st">"example_epiRomics_BW_sheet_user_paths.csv"</span>,
    package <span class="op">=</span> <span class="st">"epiRomics"</span>
  <span class="op">)</span>
<span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="fu"><a href="../reference/epiRomics_track_layer_human.html">epiRomics_track_layer_human</a></span><span class="op">(</span>
  <span class="va">epiRomics_putative_enhanceosome_fantom</span>,
  epiRomics_index <span class="op">=</span>
    <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span>
      <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">epiRomics_putative_enhanceosome_fantom</span><span class="op">@</span><span class="va">annotations</span><span class="op">)</span>
      <span class="op">==</span> <span class="fl">183</span>
    <span class="op">)</span>,
  epiRomics_dB <span class="op">=</span> <span class="va">epiRomics_dB</span>,
  epiRomics_track_connection <span class="op">=</span>
    <span class="va">epiRomics_track_connection</span>
<span class="op">)</span>
<span class="co">#&gt; [1] "not empty"</span>
<span class="co">#&gt; [1] 103.678</span>
<span class="co">#&gt; [1] "not empty"</span>
<span class="co">#&gt; [1] 77.5726</span>
<span class="co">#&gt; [1] "not empty"</span>
<span class="co">#&gt; [1] 33.2945</span>
<span class="co">#&gt; [1] "not empty"</span>
<span class="co">#&gt; [1] 50.4959</span></code></pre></div>
<p><img src="getting-started-with-epiRomics_files/figure-html/fig2-1.png" width="100%" style="display: block; margin: auto;"></p>
<p>What about a region that overlapped with active enhancers from the human islet regulome database?</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">epiRomics_putative_enhanceosome_regulome_active</span> <span class="op">&lt;-</span>
  <span class="fu"><a href="../reference/epiRomics_enhanceosome.html">epiRomics_enhanceosome</a></span><span class="op">(</span><span class="va">epiRomics_putative_enhancers_filtered_regulome_active</span>,
                         <span class="va">epiRomics_dB</span><span class="op">)</span>
<span class="co">#&gt; &gt;&gt; preparing features information...      2021-12-06 02:38:32 </span>
<span class="co">#&gt; &gt;&gt; identifying nearest features...        2021-12-06 02:38:32 </span>
<span class="co">#&gt; &gt;&gt; calculating distance from peak to TSS...   2021-12-06 02:38:33 </span>
<span class="co">#&gt; &gt;&gt; assigning genomic annotation...        2021-12-06 02:38:33 </span>
<span class="co">#&gt; &gt;&gt; adding gene annotation...          2021-12-06 02:38:39</span>
<span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span>
<span class="co">#&gt; &gt;&gt; assigning chromosome lengths           2021-12-06 02:38:39 </span>
<span class="co">#&gt; &gt;&gt; done...                    2021-12-06 02:38:39</span>

<span class="va">epiRomics_putative_enhanceosome_regulome_active</span><span class="op">@</span><span class="va">annotations</span>
<span class="co">#&gt; GRanges object with 6025 ranges and 19 metadata columns:</span>
<span class="co">#&gt;        seqnames              ranges strand |     foxa2      mafb    nkx2_2</span>
<span class="co">#&gt;           &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;</span>
<span class="co">#&gt;    456     chr1 154418514-154419684      * |         2         2         1</span>
<span class="co">#&gt;   2082     chr7     1555599-1556082      * |         1         1         1</span>
<span class="co">#&gt;   2572     chr9     2242369-2242873      * |         2         1         1</span>
<span class="co">#&gt;   3421    chr11   65416576-65419753      * |         1         1         2</span>
<span class="co">#&gt;   4709    chr17     7887867-7889135      * |         1         0         2</span>
<span class="co">#&gt;    ...      ...                 ...    ... .       ...       ...       ...</span>
<span class="co">#&gt;   5999     chrX            49184194      * |         0         0         0</span>
<span class="co">#&gt;   6001     chrX   70478965-70479351      * |         0         0         0</span>
<span class="co">#&gt;   6006     chrX 107710676-107711066      * |         0         0         0</span>
<span class="co">#&gt;   6007     chrX 107711430-107711673      * |         0         0         0</span>
<span class="co">#&gt;   6020     chrX 150874055-150874330      * |         0         0         0</span>
<span class="co">#&gt;           nxk6_1      pdx1      h2az ChIP_Hits             annotation   geneChr</span>
<span class="co">#&gt;        &lt;integer&gt; &lt;integer&gt; &lt;integer&gt; &lt;numeric&gt;            &lt;character&gt; &lt;integer&gt;</span>
<span class="co">#&gt;    456         2         1         2        10 Intron (ENST00000622..         1</span>
<span class="co">#&gt;   2082         2         2         1         8       Promoter (&lt;=1kb)         7</span>
<span class="co">#&gt;   2572         1         2         1         8      Distal Intergenic         9</span>
<span class="co">#&gt;   3421         1         1         2         8      Distal Intergenic        11</span>
<span class="co">#&gt;   4709         2         2         1         8       Promoter (&lt;=1kb)        17</span>
<span class="co">#&gt;    ...       ...       ...       ...       ...                    ...       ...</span>
<span class="co">#&gt;   5999         0         0         0         0       Promoter (&lt;=1kb)        23</span>
<span class="co">#&gt;   6001         0         0         0         0       Promoter (&lt;=1kb)        23</span>
<span class="co">#&gt;   6006         0         0         0         0      Distal Intergenic        23</span>
<span class="co">#&gt;   6007         0         0         0         0      Distal Intergenic        23</span>
<span class="co">#&gt;   6020         0         0         0         0 Intron (ENST00000370..        23</span>
<span class="co">#&gt;        geneStart   geneEnd geneLength geneStrand      geneId       transcriptId</span>
<span class="co">#&gt;        &lt;integer&gt; &lt;integer&gt;  &lt;integer&gt;  &lt;integer&gt; &lt;character&gt;        &lt;character&gt;</span>
<span class="co">#&gt;    456 154429343 154449979      20637          1        3570  ENST00000476006.5</span>
<span class="co">#&gt;   2082   1550305   1556120       5816          2      202915  ENST00000441933.5</span>
<span class="co">#&gt;   2572   2181571   2186183       4613          1        6595  ENST00000635392.1</span>
<span class="co">#&gt;   3421  65422798  65445540      22743          1      283131  ENST00000501122.2</span>
<span class="co">#&gt;   4709   7888789   7912755      23967          1        1107 ENST00000330494.12</span>
<span class="co">#&gt;    ...       ...       ...        ...        ...         ...                ...</span>
<span class="co">#&gt;   5999  49175621  49184789       9169          2        4007  ENST00000453382.5</span>
<span class="co">#&gt;   6001  70479118  70499903      20786          1        1741  ENST00000466140.5</span>
<span class="co">#&gt;   6006 107714677 107716401       1725          2        1831  ENST00000486554.1</span>
<span class="co">#&gt;   6007 107714677 107716401       1725          2        1831  ENST00000486554.1</span>
<span class="co">#&gt;   6020 150814900 150898609      83710          2       83692  ENST00000491877.1</span>
<span class="co">#&gt;        distanceToTSS         ENSEMBL      SYMBOL               GENENAME</span>
<span class="co">#&gt;            &lt;numeric&gt;     &lt;character&gt; &lt;character&gt;            &lt;character&gt;</span>
<span class="co">#&gt;    456         -9659 ENSG00000160712        IL6R interleukin 6 receptor</span>
<span class="co">#&gt;   2082            38 ENSG00000164855    TMEM184A transmembrane protei..</span>
<span class="co">#&gt;   2572         60798 ENSG00000080503     SMARCA2 SWI/SNF related, mat..</span>
<span class="co">#&gt;   3421         -3045 ENSG00000245532       NEAT1 nuclear paraspeckle ..</span>
<span class="co">#&gt;   4709             0 ENSG00000170004        CHD3 chromodomain helicas..</span>
<span class="co">#&gt;    ...           ...             ...         ...                    ...</span>
<span class="co">#&gt;   5999           595 ENSG00000012211    PRICKLE3 prickle planar cell ..</span>
<span class="co">#&gt;   6001             0 ENSG00000082458        DLG3 discs large MAGUK sc..</span>
<span class="co">#&gt;   6006          5335 ENSG00000157514     TSC22D3 TSC22 domain family ..</span>
<span class="co">#&gt;   6007          4728 ENSG00000157514     TSC22D3 TSC22 domain family ..</span>
<span class="co">#&gt;   6020         24279 ENSG00000102181      CD99L2   CD99 molecule like 2</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 640 sequences (1 circular) from hg38 genome</span>

<span class="co">## Evaluate calls on chromosome 1</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span>
  <span class="va">epiRomics_putative_enhanceosome_regulome_active</span><span class="op">@</span><span class="va">annotations</span>
<span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">epiRomics_putative_enhanceosome_regulome_active</span><span class="op">@</span><span class="va">annotations</span><span class="op">)</span><span class="op">$</span><span class="va">seqnames</span> 
  <span class="op">==</span> <span class="st">"chr1"</span>,<span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;     seqnames     start       end width strand foxa2 mafb nkx2_2 nxk6_1 pdx1</span>
<span class="co">#&gt; 456     chr1 154418514 154419684  1171      *     2    2      1      2    1</span>
<span class="co">#&gt; 82      chr1  10685395  10688670  3276      *     1    0      1      1    2</span>
<span class="co">#&gt; 46      chr1   7574092   7574479   388      *     1    1      1      1    1</span>
<span class="co">#&gt; 47      chr1   7574640   7575094   455      *     1    1      1      1    1</span>
<span class="co">#&gt; 49      chr1   8169274   8169689   416      *     1    1      1      1    1</span>
<span class="co">#&gt; 50      chr1   8170112   8170857   746      *     1    1      1      1    1</span>
<span class="co">#&gt;     h2az ChIP_Hits                                        annotation geneChr</span>
<span class="co">#&gt; 456    2        10    Intron (ENST00000622330.4/3570, intron 1 of 6)       1</span>
<span class="co">#&gt; 82     2         7  Intron (ENST00000377022.8/54897, intron 4 of 20)       1</span>
<span class="co">#&gt; 46     1         6 Intron (ENST00000303635.12/23261, intron 6 of 22)       1</span>
<span class="co">#&gt; 47     1         6 Intron (ENST00000303635.12/23261, intron 6 of 22)       1</span>
<span class="co">#&gt; 49     1         6                                 Distal Intergenic       1</span>
<span class="co">#&gt; 50     1         6                                 Distal Intergenic       1</span>
<span class="co">#&gt;     geneStart   geneEnd geneLength geneStrand    geneId      transcriptId</span>
<span class="co">#&gt; 456 154429343 154449979      20637          1      3570 ENST00000476006.5</span>
<span class="co">#&gt; 82   10660737  10693912      33176          2     54897 ENST00000478728.2</span>
<span class="co">#&gt; 46    7441096   7441554        459          2 101928836 ENST00000440032.1</span>
<span class="co">#&gt; 47    7441096   7441554        459          2 101928836 ENST00000440032.1</span>
<span class="co">#&gt; 49    8201518   8215207      13690          1 102724539 ENST00000670361.1</span>
<span class="co">#&gt; 50    8201518   8215207      13690          1 102724539 ENST00000670361.1</span>
<span class="co">#&gt;     distanceToTSS         ENSEMBL     SYMBOL</span>
<span class="co">#&gt; 456         -9659 ENSG00000160712       IL6R</span>
<span class="co">#&gt; 82           5242 ENSG00000130940      CASZ1</span>
<span class="co">#&gt; 46        -132538            &lt;NA&gt; CAMTA1-AS1</span>
<span class="co">#&gt; 47        -133086            &lt;NA&gt; CAMTA1-AS1</span>
<span class="co">#&gt; 49         -31829 ENSG00000227634  LINC01714</span>
<span class="co">#&gt; 50         -30661 ENSG00000227634  LINC01714</span>
<span class="co">#&gt;                                        GENENAME</span>
<span class="co">#&gt; 456                      interleukin 6 receptor</span>
<span class="co">#&gt; 82                         castor zinc finger 1</span>
<span class="co">#&gt; 46                       CAMTA1 antisense RNA 1</span>
<span class="co">#&gt; 47                       CAMTA1 antisense RNA 1</span>
<span class="co">#&gt; 49  long intergenic non-protein coding RNA 1714</span>
<span class="co">#&gt; 50  long intergenic non-protein coding RNA 1714</span>

<span class="co">## Find Index</span>

<span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">epiRomics_putative_enhanceosome_regulome_active</span><span class="op">@</span><span class="va">annotations</span><span class="op">)</span> <span class="op">==</span> <span class="fl">82</span><span class="op">)</span>
<span class="co">#&gt; [1] 7</span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="fu"><a href="../reference/epiRomics_track_layer_human.html">epiRomics_track_layer_human</a></span><span class="op">(</span>
  <span class="va">epiRomics_putative_enhanceosome_regulome_active</span>,
  epiRomics_index <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span>
    <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span>
      <span class="va">epiRomics_putative_enhanceosome_regulome_active</span><span class="op">@</span><span class="va">annotations</span>
    <span class="op">)</span> <span class="op">==</span> <span class="fl">82</span>
  <span class="op">)</span>,
  epiRomics_dB <span class="op">=</span> <span class="va">epiRomics_dB</span>,
  epiRomics_track_connection <span class="op">=</span> <span class="va">epiRomics_track_connection</span>
<span class="op">)</span>
<span class="co">#&gt; [1] "not empty"</span>
<span class="co">#&gt; [1] 36.1883</span>
<span class="co">#&gt; [1] "not empty"</span>
<span class="co">#&gt; [1] 14.7321</span>
<span class="co">#&gt; [1] "not empty"</span>
<span class="co">#&gt; [1] 5.30505</span>
<span class="co">#&gt; [1] "not empty"</span>
<span class="co">#&gt; [1] 2.40628</span></code></pre></div>
<p><img src="getting-started-with-epiRomics_files/figure-html/fig3-1.png" width="100%" style="display: block; margin: auto;"></p>
<p>What about a region that overlapped with super enhancers from the human islet regulome database?</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">epiRomics_putative_enhanceosome_regulome_super</span> <span class="op">&lt;-</span>
  <span class="fu"><a href="../reference/epiRomics_enhanceosome.html">epiRomics_enhanceosome</a></span><span class="op">(</span><span class="va">epiRomics_putative_enhancers_filtered_regulome_super</span>,
                         <span class="va">epiRomics_dB</span><span class="op">)</span>
<span class="co">#&gt; &gt;&gt; preparing features information...      2021-12-06 02:40:08 </span>
<span class="co">#&gt; &gt;&gt; identifying nearest features...        2021-12-06 02:40:08 </span>
<span class="co">#&gt; &gt;&gt; calculating distance from peak to TSS...   2021-12-06 02:40:08 </span>
<span class="co">#&gt; &gt;&gt; assigning genomic annotation...        2021-12-06 02:40:08 </span>
<span class="co">#&gt; &gt;&gt; adding gene annotation...          2021-12-06 02:40:14</span>
<span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span>
<span class="co">#&gt; &gt;&gt; assigning chromosome lengths           2021-12-06 02:40:14 </span>
<span class="co">#&gt; &gt;&gt; done...                    2021-12-06 02:40:14</span>

<span class="va">epiRomics_putative_enhanceosome_regulome_super</span><span class="op">@</span><span class="va">annotations</span>
<span class="co">#&gt; GRanges object with 2401 ranges and 19 metadata columns:</span>
<span class="co">#&gt;        seqnames              ranges strand |     foxa2      mafb    nkx2_2</span>
<span class="co">#&gt;           &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;</span>
<span class="co">#&gt;    166     chr1 154418514-154419684      * |         2         2         1</span>
<span class="co">#&gt;    966     chr9     2242369-2242873      * |         2         1         1</span>
<span class="co">#&gt;   1422    chr11   65416576-65419753      * |         1         1         2</span>
<span class="co">#&gt;     16     chr1   10685395-10688670      * |         1         0         1</span>
<span class="co">#&gt;    764     chr6   30748438-30749427      * |         2         1         1</span>
<span class="co">#&gt;    ...      ...                 ...    ... .       ...       ...       ...</span>
<span class="co">#&gt;   2393    chr22   46103687-46105878      * |         0         0         0</span>
<span class="co">#&gt;   2396    chr22   46108912-46109315      * |         0         0         0</span>
<span class="co">#&gt;   2397    chr22   46109916-46110442      * |         0         0         0</span>
<span class="co">#&gt;   2398    chr22   46115774-46116154      * |         0         0         0</span>
<span class="co">#&gt;   2399    chr22   46116326-46116501      * |         0         0         0</span>
<span class="co">#&gt;           nxk6_1      pdx1      h2az ChIP_Hits             annotation   geneChr</span>
<span class="co">#&gt;        &lt;integer&gt; &lt;integer&gt; &lt;integer&gt; &lt;numeric&gt;            &lt;character&gt; &lt;integer&gt;</span>
<span class="co">#&gt;    166         2         1         2        10 Intron (ENST00000622..         1</span>
<span class="co">#&gt;    966         1         2         1         8      Distal Intergenic         9</span>
<span class="co">#&gt;   1422         1         1         2         8      Distal Intergenic        11</span>
<span class="co">#&gt;     16         1         2         2         7 Intron (ENST00000377..         1</span>
<span class="co">#&gt;    764         1         1         1         7 Intron (ENST00000656..         6</span>
<span class="co">#&gt;    ...       ...       ...       ...       ...                    ...       ...</span>
<span class="co">#&gt;   2393         0         0         0         0 Exon (ENST0000038105..        22</span>
<span class="co">#&gt;   2396         0         0         0         0 Exon (ENST0000043543..        22</span>
<span class="co">#&gt;   2397         0         0         0         0       Promoter (2-3kb)        22</span>
<span class="co">#&gt;   2398         0         0         0         0       Promoter (2-3kb)        22</span>
<span class="co">#&gt;   2399         0         0         0         0       Promoter (2-3kb)        22</span>
<span class="co">#&gt;        geneStart   geneEnd geneLength geneStrand      geneId      transcriptId</span>
<span class="co">#&gt;        &lt;integer&gt; &lt;integer&gt;  &lt;integer&gt;  &lt;integer&gt; &lt;character&gt;       &lt;character&gt;</span>
<span class="co">#&gt;    166 154429343 154449979      20637          1        3570 ENST00000476006.5</span>
<span class="co">#&gt;    966   2181571   2186183       4613          1        6595 ENST00000635392.1</span>
<span class="co">#&gt;   1422  65422798  65445540      22743          1      283131 ENST00000501122.2</span>
<span class="co">#&gt;     16  10660737  10693912      33176          2       54897 ENST00000478728.2</span>
<span class="co">#&gt;    764  30743199  30744547       1349          2        8870 ENST00000259874.6</span>
<span class="co">#&gt;    ...       ...       ...        ...        ...         ...               ...</span>
<span class="co">#&gt;   2393  46112749  46112822         74          1      406883 ENST00000362116.3</span>
<span class="co">#&gt;   2396  46112749  46112822         74          1      406883 ENST00000362116.3</span>
<span class="co">#&gt;   2397  46112749  46112822         74          1      406883 ENST00000362116.3</span>
<span class="co">#&gt;   2398  46113686  46113768         83          1      406884 ENST00000385140.1</span>
<span class="co">#&gt;   2399  46113686  46113768         83          1      406884 ENST00000385140.1</span>
<span class="co">#&gt;        distanceToTSS         ENSEMBL      SYMBOL               GENENAME</span>
<span class="co">#&gt;            &lt;numeric&gt;     &lt;character&gt; &lt;character&gt;            &lt;character&gt;</span>
<span class="co">#&gt;    166         -9659 ENSG00000160712        IL6R interleukin 6 receptor</span>
<span class="co">#&gt;    966         60798 ENSG00000080503     SMARCA2 SWI/SNF related, mat..</span>
<span class="co">#&gt;   1422         -3045 ENSG00000245532       NEAT1 nuclear paraspeckle ..</span>
<span class="co">#&gt;     16          5242 ENSG00000130940       CASZ1   castor zinc finger 1</span>
<span class="co">#&gt;    764         -3891 ENSG00000137331        IER3 immediate early resp..</span>
<span class="co">#&gt;    ...           ...             ...         ...                    ...</span>
<span class="co">#&gt;   2393         -6871 ENSG00000283990   MIRLET7A3      microRNA let-7a-3</span>
<span class="co">#&gt;   2396         -3434 ENSG00000283990   MIRLET7A3      microRNA let-7a-3</span>
<span class="co">#&gt;   2397         -2307 ENSG00000283990   MIRLET7A3      microRNA let-7a-3</span>
<span class="co">#&gt;   2398          2088 ENSG00000284520    MIRLET7B        microRNA let-7b</span>
<span class="co">#&gt;   2399          2640 ENSG00000284520    MIRLET7B        microRNA let-7b</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 640 sequences (1 circular) from hg38 genome</span>

<span class="co">## Evaluate calls on chromosome 1</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span>
  <span class="va">epiRomics_putative_enhanceosome_regulome_super</span><span class="op">@</span><span class="va">annotations</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span>
    <span class="va">epiRomics_putative_enhanceosome_regulome_super</span><span class="op">@</span><span class="va">annotations</span><span class="op">)</span><span class="op">$</span><span class="va">seqnames</span> <span class="op">==</span> <span class="st">"chr1"</span>,<span class="op">]</span><span class="op">)</span>
<span class="co">#&gt;     seqnames     start       end width strand foxa2 mafb nkx2_2 nxk6_1 pdx1</span>
<span class="co">#&gt; 166     chr1 154418514 154419684  1171      *     2    2      1      2    1</span>
<span class="co">#&gt; 16      chr1  10685395  10688670  3276      *     1    0      1      1    2</span>
<span class="co">#&gt; 1       chr1   7574092   7574479   388      *     1    1      1      1    1</span>
<span class="co">#&gt; 2       chr1   7574640   7575094   455      *     1    1      1      1    1</span>
<span class="co">#&gt; 3       chr1   8169274   8169689   416      *     1    1      1      1    1</span>
<span class="co">#&gt; 4       chr1   8170112   8170857   746      *     1    1      1      1    1</span>
<span class="co">#&gt;     h2az ChIP_Hits                                        annotation geneChr</span>
<span class="co">#&gt; 166    2        10    Intron (ENST00000622330.4/3570, intron 1 of 6)       1</span>
<span class="co">#&gt; 16     2         7  Intron (ENST00000377022.8/54897, intron 4 of 20)       1</span>
<span class="co">#&gt; 1      1         6 Intron (ENST00000303635.12/23261, intron 6 of 22)       1</span>
<span class="co">#&gt; 2      1         6 Intron (ENST00000303635.12/23261, intron 6 of 22)       1</span>
<span class="co">#&gt; 3      1         6                                 Distal Intergenic       1</span>
<span class="co">#&gt; 4      1         6                                 Distal Intergenic       1</span>
<span class="co">#&gt;     geneStart   geneEnd geneLength geneStrand    geneId      transcriptId</span>
<span class="co">#&gt; 166 154429343 154449979      20637          1      3570 ENST00000476006.5</span>
<span class="co">#&gt; 16   10660737  10693912      33176          2     54897 ENST00000478728.2</span>
<span class="co">#&gt; 1     7441096   7441554        459          2 101928836 ENST00000440032.1</span>
<span class="co">#&gt; 2     7441096   7441554        459          2 101928836 ENST00000440032.1</span>
<span class="co">#&gt; 3     8201518   8215207      13690          1 102724539 ENST00000670361.1</span>
<span class="co">#&gt; 4     8201518   8215207      13690          1 102724539 ENST00000670361.1</span>
<span class="co">#&gt;     distanceToTSS         ENSEMBL     SYMBOL</span>
<span class="co">#&gt; 166         -9659 ENSG00000160712       IL6R</span>
<span class="co">#&gt; 16           5242 ENSG00000130940      CASZ1</span>
<span class="co">#&gt; 1         -132538            &lt;NA&gt; CAMTA1-AS1</span>
<span class="co">#&gt; 2         -133086            &lt;NA&gt; CAMTA1-AS1</span>
<span class="co">#&gt; 3          -31829 ENSG00000227634  LINC01714</span>
<span class="co">#&gt; 4          -30661 ENSG00000227634  LINC01714</span>
<span class="co">#&gt;                                        GENENAME</span>
<span class="co">#&gt; 166                      interleukin 6 receptor</span>
<span class="co">#&gt; 16                         castor zinc finger 1</span>
<span class="co">#&gt; 1                        CAMTA1 antisense RNA 1</span>
<span class="co">#&gt; 2                        CAMTA1 antisense RNA 1</span>
<span class="co">#&gt; 3   long intergenic non-protein coding RNA 1714</span>
<span class="co">#&gt; 4   long intergenic non-protein coding RNA 1714</span>

<span class="co">## Find Index</span>

<span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">epiRomics_putative_enhanceosome_regulome_super</span><span class="op">@</span><span class="va">annotations</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span>
<span class="co">#&gt; [1] 14</span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="fu"><a href="../reference/epiRomics_track_layer_human.html">epiRomics_track_layer_human</a></span><span class="op">(</span>
  <span class="va">epiRomics_putative_enhanceosome_regulome_super</span>,
  epiRomics_index <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span>
    <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">epiRomics_putative_enhanceosome_regulome_super</span><span class="op">@</span><span class="va">annotations</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span>
  <span class="op">)</span>,
  epiRomics_dB <span class="op">=</span> <span class="va">epiRomics_dB</span>,
  epiRomics_track_connection <span class="op">=</span> <span class="va">epiRomics_track_connection</span>
<span class="op">)</span>
<span class="co">#&gt; [1] "not empty"</span>
<span class="co">#&gt; [1] 243.743</span>
<span class="co">#&gt; [1] "not empty"</span>
<span class="co">#&gt; [1] 103.706</span>
<span class="co">#&gt; [1] "not empty"</span>
<span class="co">#&gt; [1] 11.3213</span>
<span class="co">#&gt; [1] "not empty"</span>
<span class="co">#&gt; [1] 22.1498</span></code></pre></div>
<p><img src="getting-started-with-epiRomics_files/figure-html/fig4-1.png" width="100%" style="display: block; margin: auto;"> Or, about a region that overlapped with ultra-conserved non coding elements?</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">epiRomics_putative_enhanceosome_ucnes</span> <span class="op">&lt;-</span>
  <span class="fu"><a href="../reference/epiRomics_enhanceosome.html">epiRomics_enhanceosome</a></span><span class="op">(</span><span class="va">epiRomics_putative_enhancers_filtered_ucnes</span>, <span class="va">epiRomics_dB</span><span class="op">)</span>
<span class="co">#&gt; &gt;&gt; preparing features information...      2021-12-06 02:41:50 </span>
<span class="co">#&gt; &gt;&gt; identifying nearest features...        2021-12-06 02:41:50 </span>
<span class="co">#&gt; &gt;&gt; calculating distance from peak to TSS...   2021-12-06 02:41:51 </span>
<span class="co">#&gt; &gt;&gt; assigning genomic annotation...        2021-12-06 02:41:51 </span>
<span class="co">#&gt; &gt;&gt; adding gene annotation...          2021-12-06 02:41:57</span>
<span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span>
<span class="co">#&gt; &gt;&gt; assigning chromosome lengths           2021-12-06 02:41:57 </span>
<span class="co">#&gt; &gt;&gt; done...                    2021-12-06 02:41:57</span>

<span class="va">epiRomics_putative_enhanceosome_ucnes</span><span class="op">@</span><span class="va">annotations</span>
<span class="co">#&gt; GRanges object with 11 ranges and 19 metadata columns:</span>
<span class="co">#&gt;      seqnames              ranges strand |     foxa2      mafb    nkx2_2</span>
<span class="co">#&gt;         &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;</span>
<span class="co">#&gt;    7     chr9 106921420-106921764      * |         1         1         1</span>
<span class="co">#&gt;    1     chr1 164635220-164635921      * |         0         0         2</span>
<span class="co">#&gt;    6     chr3   71131859-71132164      * |         1         0         0</span>
<span class="co">#&gt;    2     chr1 164711914-164712296      * |         0         0         0</span>
<span class="co">#&gt;    3     chr1 164712350-164713071      * |         0         0         0</span>
<span class="co">#&gt;    8    chr11 114163425-114164860      * |         0         0         0</span>
<span class="co">#&gt;    9    chr15   36903894-36904085      * |         0         0         1</span>
<span class="co">#&gt;   11    chr21   16534340-16534665      * |         0         0         0</span>
<span class="co">#&gt;    4     chr1 200079185-200079426      * |         0         0         0</span>
<span class="co">#&gt;    5     chr1 213585694-213586385      * |         0         0         0</span>
<span class="co">#&gt;   10    chr15   53447393-53447809      * |         0         0         0</span>
<span class="co">#&gt;         nxk6_1      pdx1      h2az ChIP_Hits             annotation   geneChr</span>
<span class="co">#&gt;      &lt;integer&gt; &lt;integer&gt; &lt;integer&gt; &lt;numeric&gt;            &lt;character&gt; &lt;integer&gt;</span>
<span class="co">#&gt;    7         1         0         1         5 Intron (ENST00000472..         9</span>
<span class="co">#&gt;    1         0         1         1         4 Intron (ENST00000420..         1</span>
<span class="co">#&gt;    6         0         0         1         2       Promoter (&lt;=1kb)         3</span>
<span class="co">#&gt;    2         0         0         1         1 Intron (ENST00000420..         1</span>
<span class="co">#&gt;    3         0         0         1         1 Intron (ENST00000420..         1</span>
<span class="co">#&gt;    8         0         0         1         1       Promoter (2-3kb)        11</span>
<span class="co">#&gt;    9         0         0         0         1       Promoter (&lt;=1kb)        15</span>
<span class="co">#&gt;   11         0         0         1         1       Promoter (&lt;=1kb)        21</span>
<span class="co">#&gt;    4         0         0         0         0 Intron (ENST00000236..         1</span>
<span class="co">#&gt;    5         0         0         0         0      Distal Intergenic         1</span>
<span class="co">#&gt;   10         0         0         0         0 Intron (ENST00000662..        15</span>
<span class="co">#&gt;      geneStart   geneEnd geneLength geneStrand      geneId      transcriptId</span>
<span class="co">#&gt;      &lt;integer&gt; &lt;integer&gt;  &lt;integer&gt;  &lt;integer&gt; &lt;character&gt;       &lt;character&gt;</span>
<span class="co">#&gt;    7 106926925 106932462       5538          1       58499 ENST00000480607.5</span>
<span class="co">#&gt;    1 164630981 164799889     168909          1        5087 ENST00000482110.5</span>
<span class="co">#&gt;    6  70959237  71132099     172863          2       27086 ENST00000650188.1</span>
<span class="co">#&gt;    2 164772912 164807571      34660          1        5087 ENST00000558837.5</span>
<span class="co">#&gt;    3 164772912 164807571      34660          1        5087 ENST00000558837.5</span>
<span class="co">#&gt;    8 114159791 114161166       1376          2   101928875 ENST00000669692.1</span>
<span class="co">#&gt;    9  36894784  36904067       9284          2        4212 ENST00000559408.1</span>
<span class="co">#&gt;   11  16534952  16607137      72186          1      388815 ENST00000654245.1</span>
<span class="co">#&gt;    4 200043810 200058424      14615          1        2494 ENST00000367357.3</span>
<span class="co">#&gt;    5 213492288 213579267      86980          1   105372912 ENST00000417161.2</span>
<span class="co">#&gt;   10  53419865  53453487      33623          1   105370826 ENST00000662193.1</span>
<span class="co">#&gt;      distanceToTSS         ENSEMBL       SYMBOL               GENENAME</span>
<span class="co">#&gt;          &lt;numeric&gt;     &lt;character&gt;  &lt;character&gt;            &lt;character&gt;</span>
<span class="co">#&gt;    7         -5161 ENSG00000148143       ZNF462 zinc finger protein ..</span>
<span class="co">#&gt;    1          4239 ENSG00000185630         PBX1         PBX homeobox 1</span>
<span class="co">#&gt;    6             0 ENSG00000114861        FOXP1        forkhead box P1</span>
<span class="co">#&gt;    2        -60616 ENSG00000185630         PBX1         PBX homeobox 1</span>
<span class="co">#&gt;    3        -59841 ENSG00000185630         PBX1         PBX homeobox 1</span>
<span class="co">#&gt;    8         -2259            &lt;NA&gt; LOC101928875 uncharacterized LOC1..</span>
<span class="co">#&gt;    9             0 ENSG00000134138        MEIS2        Meis homeobox 2</span>
<span class="co">#&gt;   11          -287 ENSG00000215386     MIR99AHG mir-99a-let-7c clust..</span>
<span class="co">#&gt;    4         35375 ENSG00000116833        NR5A2 nuclear receptor sub..</span>
<span class="co">#&gt;    5         93406            &lt;NA&gt; LOC105372912 uncharacterized LOC1..</span>
<span class="co">#&gt;   10         27528            &lt;NA&gt; LOC105370826 uncharacterized LOC1..</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 640 sequences (1 circular) from hg38 genome</span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="fu"><a href="../reference/epiRomics_track_layer_human.html">epiRomics_track_layer_human</a></span><span class="op">(</span>
  <span class="va">epiRomics_putative_enhanceosome_ucnes</span>,
  epiRomics_index <span class="op">=</span> <span class="fl">9</span>,
  epiRomics_dB <span class="op">=</span> <span class="va">epiRomics_dB</span>,
  epiRomics_track_connection <span class="op">=</span> <span class="va">epiRomics_track_connection</span>
<span class="op">)</span>
<span class="co">#&gt; [1] "not empty"</span>
<span class="co">#&gt; [1] 82.795</span>
<span class="co">#&gt; [1] "not empty"</span>
<span class="co">#&gt; [1] 62.847</span>
<span class="co">#&gt; [1] "not empty"</span>
<span class="co">#&gt; [1] 3.55117</span>
<span class="co">#&gt; [1] "not empty"</span>
<span class="co">#&gt; [1] 3.80884</span></code></pre></div>
<p><img src="getting-started-with-epiRomics_files/figure-html/fig5-1.png" width="100%" style="display: block; margin: auto;"></p>
<p>How about applying multiple filters to further increase the confidence of calls?</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">epiRomics_putative_enhancers_filtered_stringent</span> <span class="op">&lt;-</span>
  <span class="fu"><a href="../reference/epiRomics_enhancers_filter.html">epiRomics_enhancers_filter</a></span><span class="op">(</span>
    <span class="fu"><a href="../reference/epiRomics_enhancers_filter.html">epiRomics_enhancers_filter</a></span><span class="op">(</span>
      <span class="fu"><a href="../reference/epiRomics_enhancers_filter.html">epiRomics_enhancers_filter</a></span><span class="op">(</span>
        <span class="fu"><a href="../reference/epiRomics_enhancers_filter.html">epiRomics_enhancers_filter</a></span><span class="op">(</span><span class="va">epiRomics_putative_enhancers</span>, <span class="va">epiRomics_dB</span>,
                                   epiRomics_type <span class="op">=</span>
                                     <span class="st">"hg38_custom_fantom"</span><span class="op">)</span>,
        <span class="va">epiRomics_dB</span>,
        epiRomics_type <span class="op">=</span> <span class="st">"hg38_custom_regulome_active"</span>
      <span class="op">)</span>,
      <span class="va">epiRomics_dB</span>,
      epiRomics_type <span class="op">=</span> <span class="st">"hg38_custom_regulome_super"</span>
    <span class="op">)</span>,
    <span class="va">epiRomics_dB</span>,
    epiRomics_type <span class="op">=</span> <span class="st">"hg38_custom_ucnes"</span>
  <span class="op">)</span>


<span class="co">## Here, we see a highly conservative list of putative enhancer calls that overlap with four</span>
<span class="co">## different functional annotations, suggesting the lowest hanging fruit for downstream </span>
<span class="co">## bench-lab validation. </span>
<span class="co">## NOTE: The UCNE database filter caused the greatest reduction in enhancer calls.</span>

<span class="va">epiRomics_putative_enhancers_filtered_stringent</span><span class="op">@</span><span class="va">annotations</span>
<span class="co">#&gt; GRanges object with 2 ranges and 0 metadata columns:</span>
<span class="co">#&gt;       seqnames              ranges strand</span>
<span class="co">#&gt;          &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;   [1]     chr1 164711914-164712296      *</span>
<span class="co">#&gt;   [2]     chr1 164712350-164713071      *</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 640 sequences (1 circular) from hg38 genome</span>

<span class="va">epiRomics_putative_enhanceosome_stringent</span> <span class="op">&lt;-</span>
  <span class="fu"><a href="../reference/epiRomics_enhanceosome.html">epiRomics_enhanceosome</a></span><span class="op">(</span><span class="va">epiRomics_putative_enhancers_filtered_stringent</span>,
                         <span class="va">epiRomics_dB</span><span class="op">)</span>
<span class="co">#&gt; &gt;&gt; preparing features information...      2021-12-06 02:43:28 </span>
<span class="co">#&gt; &gt;&gt; identifying nearest features...        2021-12-06 02:43:28 </span>
<span class="co">#&gt; &gt;&gt; calculating distance from peak to TSS...   2021-12-06 02:43:29 </span>
<span class="co">#&gt; &gt;&gt; assigning genomic annotation...        2021-12-06 02:43:29 </span>
<span class="co">#&gt; &gt;&gt; adding gene annotation...          2021-12-06 02:43:35</span>
<span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span>
<span class="co">#&gt; &gt;&gt; assigning chromosome lengths           2021-12-06 02:43:35 </span>
<span class="co">#&gt; &gt;&gt; done...                    2021-12-06 02:43:35</span></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="fu"><a href="../reference/epiRomics_track_layer_human.html">epiRomics_track_layer_human</a></span><span class="op">(</span>
  <span class="va">epiRomics_putative_enhanceosome_stringent</span>,
  epiRomics_index <span class="op">=</span> <span class="fl">1</span>,
  epiRomics_dB <span class="op">=</span> <span class="va">epiRomics_dB</span>,
  epiRomics_track_connection <span class="op">=</span> <span class="va">epiRomics_track_connection</span>
<span class="op">)</span>
<span class="co">#&gt; [1] "not empty"</span>
<span class="co">#&gt; [1] 233.705</span>
<span class="co">#&gt; [1] "not empty"</span>
<span class="co">#&gt; [1] 221.083</span>
<span class="co">#&gt; [1] "not empty"</span>
<span class="co">#&gt; [1] 9.31751</span>
<span class="co">#&gt; [1] "not empty"</span>
<span class="co">#&gt; [1] 15.8259</span></code></pre></div>
<p><img src="getting-started-with-epiRomics_files/figure-html/fig6-1.png" width="100%" style="display: block; margin: auto;"></p>
<p>How can we use these putative enhanceosome regions to infer biology between cell states? In this example, we will integrate ATAC-Seq data differential testing showing differences in chromatin accessibility between alpha and beta cells</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Read differentially binding data generated with DiffBind.</span>
<span class="co">## DE comparing human alpha and beta cell chromatin.</span>

<span class="va">b.v.a</span> <span class="op">&lt;-</span>
  <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"DBA_Beta_Versus_Alpha.csv"</span>, package <span class="op">=</span> <span class="st">"epiRomics"</span><span class="op">)</span><span class="op">)</span>
<span class="va">b.v.a</span> <span class="op">&lt;-</span> <span class="fu">GRanges</span><span class="op">(</span><span class="va">b.v.a</span><span class="op">)</span>

<span class="co"># Filter for beta enriched chromatin regions</span>

<span class="va">beta.enriched</span> <span class="op">&lt;-</span> <span class="va">b.v.a</span><span class="op">[</span><span class="va">b.v.a</span><span class="op">$</span><span class="va">Fold</span> <span class="op">&gt;=</span> <span class="fl">1</span>, <span class="op">]</span>

<span class="co"># Connect to our putative enhanceosomes</span>

<span class="va">beta_enhancer_regions</span> <span class="op">&lt;-</span>
  <span class="fu"><a href="../reference/epiRomics_regions_of_interest.html">epiRomics_regions_of_interest</a></span><span class="op">(</span><span class="va">epiRomics_putative_enhanceosome_fantom</span>, <span class="va">beta.enriched</span><span class="op">)</span></code></pre></div>
<p>Now, lets visualize the top candidate region we found after connecting our differential chromatin analysis with the putative enhanceosomes</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="fu"><a href="../reference/epiRomics_track_layer_human.html">epiRomics_track_layer_human</a></span><span class="op">(</span>
  <span class="va">beta_enhancer_regions</span>,
  epiRomics_index <span class="op">=</span> <span class="fl">1</span>,
  epiRomics_dB <span class="op">=</span> <span class="va">epiRomics_dB</span>,
  epiRomics_track_connection <span class="op">=</span> <span class="va">epiRomics_track_connection</span>
<span class="op">)</span>
<span class="co">#&gt; [1] "not empty"</span>
<span class="co">#&gt; [1] 16.5139</span>
<span class="co">#&gt; [1] "not empty"</span>
<span class="co">#&gt; [1] 14.2229</span>
<span class="co">#&gt; [1] "not empty"</span>
<span class="co">#&gt; [1] 2.38379</span>
<span class="co">#&gt; [1] "not empty"</span>
<span class="co">#&gt; [1] 1.29041</span></code></pre></div>
<p><img src="getting-started-with-epiRomics_files/figure-html/fig7-1.png" width="100%" style="display: block; margin: auto;"></p>
</div>
<div class="section level1">
<h1 id="session-information">Session Information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h1>
<p>Here is the output of <code><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo()</a></code> on the system on which this document was compiled:</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.1.2 (2021-11-01)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Big Sur 10.16</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">#&gt; [8] base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] rtracklayer_1.54.0                      </span>
<span class="co">#&gt;  [2] org.Hs.eg.db_3.14.0                     </span>
<span class="co">#&gt;  [3] TxDb.Hsapiens.UCSC.hg38.knownGene_3.14.0</span>
<span class="co">#&gt;  [4] GenomicFeatures_1.46.1                  </span>
<span class="co">#&gt;  [5] AnnotationDbi_1.56.2                    </span>
<span class="co">#&gt;  [6] Biobase_2.54.0                          </span>
<span class="co">#&gt;  [7] GenomicRanges_1.46.1                    </span>
<span class="co">#&gt;  [8] GenomeInfoDb_1.30.0                     </span>
<span class="co">#&gt;  [9] IRanges_2.28.0                          </span>
<span class="co">#&gt; [10] S4Vectors_0.32.3                        </span>
<span class="co">#&gt; [11] BiocGenerics_0.40.0                     </span>
<span class="co">#&gt; [12] epiRomics_0.1.3                         </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] utf8_1.2.2                             </span>
<span class="co">#&gt;   [2] tidyselect_1.1.1                       </span>
<span class="co">#&gt;   [3] RSQLite_2.2.8                          </span>
<span class="co">#&gt;   [4] htmlwidgets_1.5.4                      </span>
<span class="co">#&gt;   [5] grid_4.1.2                             </span>
<span class="co">#&gt;   [6] BiocParallel_1.28.2                    </span>
<span class="co">#&gt;   [7] scatterpie_0.1.7                       </span>
<span class="co">#&gt;   [8] munsell_0.5.0                          </span>
<span class="co">#&gt;   [9] codetools_0.2-18                       </span>
<span class="co">#&gt;  [10] ragg_1.2.0                             </span>
<span class="co">#&gt;  [11] withr_2.4.3                            </span>
<span class="co">#&gt;  [12] colorspace_2.0-2                       </span>
<span class="co">#&gt;  [13] GOSemSim_2.20.0                        </span>
<span class="co">#&gt;  [14] filelock_1.0.2                         </span>
<span class="co">#&gt;  [15] highr_0.9                              </span>
<span class="co">#&gt;  [16] knitr_1.36                             </span>
<span class="co">#&gt;  [17] rstudioapi_0.13                        </span>
<span class="co">#&gt;  [18] DOSE_3.20.1                            </span>
<span class="co">#&gt;  [19] MatrixGenerics_1.6.0                   </span>
<span class="co">#&gt;  [20] GenomeInfoDbData_1.2.7                 </span>
<span class="co">#&gt;  [21] polyclip_1.10-0                        </span>
<span class="co">#&gt;  [22] bit64_4.0.5                            </span>
<span class="co">#&gt;  [23] farver_2.1.0                           </span>
<span class="co">#&gt;  [24] rprojroot_2.0.2                        </span>
<span class="co">#&gt;  [25] treeio_1.18.1                          </span>
<span class="co">#&gt;  [26] vctrs_0.3.8                            </span>
<span class="co">#&gt;  [27] generics_0.1.1                         </span>
<span class="co">#&gt;  [28] TH.data_1.1-0                          </span>
<span class="co">#&gt;  [29] xfun_0.28                              </span>
<span class="co">#&gt;  [30] biovizBase_1.42.0                      </span>
<span class="co">#&gt;  [31] BiocFileCache_2.2.0                    </span>
<span class="co">#&gt;  [32] party_1.3-9                            </span>
<span class="co">#&gt;  [33] regioneR_1.26.0                        </span>
<span class="co">#&gt;  [34] R6_2.5.1                               </span>
<span class="co">#&gt;  [35] graphlayouts_0.7.2                     </span>
<span class="co">#&gt;  [36] AnnotationFilter_1.18.0                </span>
<span class="co">#&gt;  [37] gridGraphics_0.5-1                     </span>
<span class="co">#&gt;  [38] bitops_1.0-7                           </span>
<span class="co">#&gt;  [39] cachem_1.0.6                           </span>
<span class="co">#&gt;  [40] fgsea_1.20.0                           </span>
<span class="co">#&gt;  [41] DelayedArray_0.20.0                    </span>
<span class="co">#&gt;  [42] assertthat_0.2.1                       </span>
<span class="co">#&gt;  [43] vroom_1.5.7                            </span>
<span class="co">#&gt;  [44] promises_1.2.0.1                       </span>
<span class="co">#&gt;  [45] BiocIO_1.4.0                           </span>
<span class="co">#&gt;  [46] scales_1.1.1                           </span>
<span class="co">#&gt;  [47] multcomp_1.4-17                        </span>
<span class="co">#&gt;  [48] ggraph_2.0.5                           </span>
<span class="co">#&gt;  [49] nnet_7.3-16                            </span>
<span class="co">#&gt;  [50] enrichplot_1.13.1.992                  </span>
<span class="co">#&gt;  [51] gtable_0.3.0                           </span>
<span class="co">#&gt;  [52] tidygraph_1.2.0                        </span>
<span class="co">#&gt;  [53] sandwich_3.0-1                         </span>
<span class="co">#&gt;  [54] ensembldb_2.18.2                       </span>
<span class="co">#&gt;  [55] rlang_0.4.12                           </span>
<span class="co">#&gt;  [56] systemfonts_1.0.3                      </span>
<span class="co">#&gt;  [57] splines_4.1.2                          </span>
<span class="co">#&gt;  [58] lazyeval_0.2.2                         </span>
<span class="co">#&gt;  [59] dichromat_2.0-0                        </span>
<span class="co">#&gt;  [60] checkmate_2.0.0                        </span>
<span class="co">#&gt;  [61] BiocManager_1.30.16                    </span>
<span class="co">#&gt;  [62] yaml_2.2.1                             </span>
<span class="co">#&gt;  [63] reshape2_1.4.4                         </span>
<span class="co">#&gt;  [64] backports_1.4.0                        </span>
<span class="co">#&gt;  [65] httpuv_1.6.3                           </span>
<span class="co">#&gt;  [66] qvalue_2.26.0                          </span>
<span class="co">#&gt;  [67] Hmisc_4.6-0                            </span>
<span class="co">#&gt;  [68] tools_4.1.2                            </span>
<span class="co">#&gt;  [69] ggplotify_0.1.0                        </span>
<span class="co">#&gt;  [70] ggplot2_3.3.5                          </span>
<span class="co">#&gt;  [71] gplots_3.1.1                           </span>
<span class="co">#&gt;  [72] ellipsis_0.3.2                         </span>
<span class="co">#&gt;  [73] jquerylib_0.1.4                        </span>
<span class="co">#&gt;  [74] RColorBrewer_1.1-2                     </span>
<span class="co">#&gt;  [75] Rcpp_1.0.7                             </span>
<span class="co">#&gt;  [76] plyr_1.8.6                             </span>
<span class="co">#&gt;  [77] base64enc_0.1-3                        </span>
<span class="co">#&gt;  [78] progress_1.2.2                         </span>
<span class="co">#&gt;  [79] zlibbioc_1.40.0                        </span>
<span class="co">#&gt;  [80] purrr_0.3.4                            </span>
<span class="co">#&gt;  [81] RCurl_1.98-1.5                         </span>
<span class="co">#&gt;  [82] prettyunits_1.1.1                      </span>
<span class="co">#&gt;  [83] rpart_4.1-15                           </span>
<span class="co">#&gt;  [84] viridis_0.6.2                          </span>
<span class="co">#&gt;  [85] zoo_1.8-9                              </span>
<span class="co">#&gt;  [86] SummarizedExperiment_1.24.0            </span>
<span class="co">#&gt;  [87] ggrepel_0.9.1                          </span>
<span class="co">#&gt;  [88] cluster_2.1.2                          </span>
<span class="co">#&gt;  [89] fs_1.5.1                               </span>
<span class="co">#&gt;  [90] magrittr_2.0.1                         </span>
<span class="co">#&gt;  [91] data.table_1.14.2                      </span>
<span class="co">#&gt;  [92] DO.db_2.9                              </span>
<span class="co">#&gt;  [93] mvtnorm_1.1-3                          </span>
<span class="co">#&gt;  [94] ProtGenerics_1.26.0                    </span>
<span class="co">#&gt;  [95] matrixStats_0.61.0                     </span>
<span class="co">#&gt;  [96] patchwork_1.1.1                        </span>
<span class="co">#&gt;  [97] hms_1.1.1                              </span>
<span class="co">#&gt;  [98] mime_0.12                              </span>
<span class="co">#&gt;  [99] evaluate_0.14                          </span>
<span class="co">#&gt; [100] xtable_1.8-4                           </span>
<span class="co">#&gt; [101] XML_3.99-0.8                           </span>
<span class="co">#&gt; [102] jpeg_0.1-9                             </span>
<span class="co">#&gt; [103] gridExtra_2.3                          </span>
<span class="co">#&gt; [104] compiler_4.1.2                         </span>
<span class="co">#&gt; [105] biomaRt_2.50.1                         </span>
<span class="co">#&gt; [106] tibble_3.1.6                           </span>
<span class="co">#&gt; [107] KernSmooth_2.23-20                     </span>
<span class="co">#&gt; [108] shadowtext_0.0.9                       </span>
<span class="co">#&gt; [109] crayon_1.4.2                           </span>
<span class="co">#&gt; [110] htmltools_0.5.2                        </span>
<span class="co">#&gt; [111] ggfun_0.0.4                            </span>
<span class="co">#&gt; [112] later_1.3.0                            </span>
<span class="co">#&gt; [113] tzdb_0.2.0                             </span>
<span class="co">#&gt; [114] Formula_1.2-4                          </span>
<span class="co">#&gt; [115] aplot_0.1.1                            </span>
<span class="co">#&gt; [116] tidyr_1.1.4                            </span>
<span class="co">#&gt; [117] libcoin_1.0-9                          </span>
<span class="co">#&gt; [118] DBI_1.1.1                              </span>
<span class="co">#&gt; [119] ChIPseeker_1.31.2                      </span>
<span class="co">#&gt; [120] tweenr_1.0.2                           </span>
<span class="co">#&gt; [121] formatR_1.11                           </span>
<span class="co">#&gt; [122] dbplyr_2.1.1                           </span>
<span class="co">#&gt; [123] MASS_7.3-54                            </span>
<span class="co">#&gt; [124] rappdirs_0.3.3                         </span>
<span class="co">#&gt; [125] boot_1.3-28                            </span>
<span class="co">#&gt; [126] Matrix_1.3-4                           </span>
<span class="co">#&gt; [127] readr_2.1.1                            </span>
<span class="co">#&gt; [128] parallel_4.1.2                         </span>
<span class="co">#&gt; [129] Gviz_1.38.0                            </span>
<span class="co">#&gt; [130] igraph_1.2.9                           </span>
<span class="co">#&gt; [131] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2</span>
<span class="co">#&gt; [132] pkgconfig_2.0.3                        </span>
<span class="co">#&gt; [133] pkgdown_2.0.1                          </span>
<span class="co">#&gt; [134] GenomicAlignments_1.30.0               </span>
<span class="co">#&gt; [135] coin_1.4-2                             </span>
<span class="co">#&gt; [136] foreign_0.8-81                         </span>
<span class="co">#&gt; [137] xml2_1.3.3                             </span>
<span class="co">#&gt; [138] ggtree_3.2.1                           </span>
<span class="co">#&gt; [139] bslib_0.3.1                            </span>
<span class="co">#&gt; [140] XVector_0.34.0                         </span>
<span class="co">#&gt; [141] yulab.utils_0.0.4                      </span>
<span class="co">#&gt; [142] stringr_1.4.0                          </span>
<span class="co">#&gt; [143] VariantAnnotation_1.40.0               </span>
<span class="co">#&gt; [144] digest_0.6.29                          </span>
<span class="co">#&gt; [145] strucchange_1.5-2                      </span>
<span class="co">#&gt; [146] Biostrings_2.62.0                      </span>
<span class="co">#&gt; [147] fastmatch_1.1-3                        </span>
<span class="co">#&gt; [148] rmarkdown_2.11                         </span>
<span class="co">#&gt; [149] tidytree_0.3.6                         </span>
<span class="co">#&gt; [150] htmlTable_2.3.0                        </span>
<span class="co">#&gt; [151] annotatr_1.20.0                        </span>
<span class="co">#&gt; [152] restfulr_0.0.13                        </span>
<span class="co">#&gt; [153] curl_4.3.2                             </span>
<span class="co">#&gt; [154] gtools_3.9.2                           </span>
<span class="co">#&gt; [155] shiny_1.7.1                            </span>
<span class="co">#&gt; [156] Rsamtools_2.10.0                       </span>
<span class="co">#&gt; [157] modeltools_0.2-23                      </span>
<span class="co">#&gt; [158] rjson_0.2.20                           </span>
<span class="co">#&gt; [159] nlme_3.1-153                           </span>
<span class="co">#&gt; [160] lifecycle_1.0.1                        </span>
<span class="co">#&gt; [161] jsonlite_1.7.2                         </span>
<span class="co">#&gt; [162] desc_1.4.0                             </span>
<span class="co">#&gt; [163] viridisLite_0.4.0                      </span>
<span class="co">#&gt; [164] BSgenome_1.62.0                        </span>
<span class="co">#&gt; [165] fansi_0.5.0                            </span>
<span class="co">#&gt; [166] pillar_1.6.4                           </span>
<span class="co">#&gt; [167] lattice_0.20-45                        </span>
<span class="co">#&gt; [168] plotrix_3.8-2                          </span>
<span class="co">#&gt; [169] KEGGREST_1.34.0                        </span>
<span class="co">#&gt; [170] fastmap_1.1.0                          </span>
<span class="co">#&gt; [171] httr_1.4.2                             </span>
<span class="co">#&gt; [172] survival_3.2-13                        </span>
<span class="co">#&gt; [173] GO.db_3.14.0                           </span>
<span class="co">#&gt; [174] interactiveDisplayBase_1.32.0          </span>
<span class="co">#&gt; [175] glue_1.5.1                             </span>
<span class="co">#&gt; [176] png_0.1-7                              </span>
<span class="co">#&gt; [177] BiocVersion_3.14.0                     </span>
<span class="co">#&gt; [178] bit_4.0.4                              </span>
<span class="co">#&gt; [179] ggforce_0.3.3                          </span>
<span class="co">#&gt; [180] stringi_1.7.6                          </span>
<span class="co">#&gt; [181] sass_0.4.0                             </span>
<span class="co">#&gt; [182] blob_1.2.2                             </span>
<span class="co">#&gt; [183] textshaping_0.3.6                      </span>
<span class="co">#&gt; [184] AnnotationHub_3.2.0                    </span>
<span class="co">#&gt; [185] caTools_1.18.2                         </span>
<span class="co">#&gt; [186] latticeExtra_0.6-29                    </span>
<span class="co">#&gt; [187] memoise_2.0.1                          </span>
<span class="co">#&gt; [188] dplyr_1.0.7                            </span>
<span class="co">#&gt; [189] ape_5.5</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Alex M. Mawla.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
